From e3947492a183a9c566dd091c7ada607d8b729c2f Mon Sep 17 00:00:00 2001
From: Andreas Baierl <ichgeh@imkreisrum.de>
Date: Mon, 8 May 2023 17:29:06 +0200
Subject: [PATCH] fix

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 251ca2f..4592e47 100644
--- a/Makefile
+++ b/Makefile
@@ -109,7 +109,7 @@ install-i18n: $(I18Nmsgs)
 ### Targets:
 
 mpg2c: mpg2c.c
-	$(CC) mpg2c.c -o $@
+	gcc mpg2c.c -o $@
 
 $(SOFILE): $(OBJS) $(OBJS_MPG)
 	$(CXX) $(CXXFLAGS) $(LDFLAGS) -shared $(OBJS) $(OBJS_MPG) -o $@
-- 
2.30.2

