diff -ru8bBwd a/Makefile b/Makefile
--- a/Makefile	2025-11-01 10:31:35.000000000 +0100
+++ b/Makefile	2025-11-04 08:01:06.660984307 +0100
@@ -173,16 +173,18 @@
 ifeq ($(DRM),1)
 PLUGIN = softhddrm
 CONFIG += -DUSE_DRM -DVAAPI
 _CFLAGS += $(shell pkg-config --cflags libdrm)
 LIBS += $(shell pkg-config --libs egl gbm libdrm)
 endif
 
 ifeq ($(CUVID),1)
+export CFLAGS	= $(call PKGCFG,cflags) -fpermissive
+export CXXFLAGS = $(call PKGCFG,cxxflags) -fpermissive
 #CONFIG += -DUSE_PIP			# PIP support
 CONFIG += -DCUVID			# enable CUVID decoder
 LIBS += $(shell pkg-config --libs egl gl)
 ifeq ($(YADIF),1)
 CONFIG += -DYADIF			# Yadif only with CUVID
 endif
 endif
 
