From 8ecf6ff230c7b7f147a40125d600fbaea172f015 Mon Sep 17 00:00:00 2001
From: Andreas Baierl <ichgeh@imkreisrum.de>
Date: Mon, 16 Sep 2024 11:16:26 +0200
Subject: [PATCH] fix makefile

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 7cfea18..710e734 100644
--- a/Makefile
+++ b/Makefile
@@ -113,8 +113,8 @@ TMPDIR ?= /tmp
 
 ### The compiler options:
 
-export CFLAGS	= $(call PKGCFG,cflags)
-export CXXFLAGS = $(call PKGCFG,cxxflags)
+export CFLAGS	= $(call PKGCFG,cflags) -fpermissive
+export CXXFLAGS = $(call PKGCFG,cxxflags) -fpermissive
 
 ifeq ($(CFLAGS),)
 $(warning CFLAGS not set)
-- 
2.39.2

