diff -ru8bBwd a/codec.c b/codec.c
--- a/codec.c	2024-01-19 11:45:54.000000000 +0100
+++ b/codec.c	2024-01-19 20:02:28.695331444 +0100
@@ -556,17 +556,17 @@
                         if (init_filters(video_ctx, decoder->HwDecoder, frame) < 0) {
                             Debug(3,"video: Init of YADIF Filter failed\n");
                             decoder->filter = 0;
                         } else {
                             Debug(3, "Init YADIF ok\n");
                             decoder->filter = 2;
                         }
                     }
-                    if (decoder->filter == 2)) { 
+                    if (decoder->filter == 2) {
                         ret = push_filters(video_ctx, decoder->HwDecoder, frame);
                         // av_frame_unref(frame);
                         continue;
                     }
                 }
 #endif
                 //  DisplayPts(video_ctx, frame);
                 VideoRenderFrame(decoder->HwDecoder, video_ctx, frame);
