diff --git a/Makefile b/Makefile
index 726019a..5d27265 100644
--- a/Makefile
+++ b/Makefile
@@ -142,8 +142,7 @@ install-lib: $(SOFILE)
 
 install-conf:
 	@mkdir -p $(DESTDIR)$(CFGDIR)/plugins/$(PLUGIN)
-	@mkdir -p $(DESTDIR)$(RESDIR)/plugins/$(PLUGIN)
-	@cp -pn $(PLUGIN)/* $(DESTDIR)$(RESDIR)/plugins/$(PLUGIN)/
+	cp -pn $(PLUGIN)/* $(DESTDIR)$(CFGDIR)/plugins/$(PLUGIN)/
 
 install: install-lib install-i18n install-conf
 
