diff -ru8bBwd a/Makefile b/Makefile
--- a/Makefile	2023-01-04 18:49:07.000000000 +0100
+++ b/Makefile	2023-01-05 17:22:45.792314581 +0100
@@ -11,17 +11,17 @@
 VERSION := $(shell awk '/#define LIVEVERSION/ { print $$3 }' setup.h | sed -e 's/[";]//g')
 
 ### Check for libpcre2
 HAVE_PCRE2 := $(shell if pkg-config --exists libpcre2-8; then echo "1"; else echo "0"; fi )
 
 ### The directory environment:
 # Use package data if installed...otherwise assume we're under the VDR source directory:
 PKGCFG = $(if $(VDRDIR),$(shell pkg-config --variable=$(1) $(VDRDIR)/vdr.pc),$(shell PKG_CONFIG_PATH="$$PKG_CONFIG_PATH:../../.." pkg-config --variable=$(1) vdr))
-LIBDIR := $(call PKGCFG,libdir)
+LIBDIR ?= $(call PKGCFG,libdir)
 LOCDIR := $(call PKGCFG,locdir)
 PLGCFG := $(call PKGCFG,plgcfg)
 RESDIR := $(call PKGCFG,resdir)
 #
 TMPDIR ?= /tmp
 
 ### The compiler options:
 export CFLAGS   := $(call PKGCFG,cflags)
@@ -74,17 +74,17 @@
 ### The name of the distribution archive:
 ARCHIVE := $(PLUGIN)-$(VERSION)
 PACKAGE := vdr-$(ARCHIVE)
 
 ### The name of the shared object file:
 SOFILE := libvdr-$(PLUGIN).so
 
 ### Installed shared object file:
-SOINST := $(DESTDIR)$(LIBDIR)/$(SOFILE).$(APIVERSION)
+SOINST := $(LIBDIR)/$(SOFILE).$(APIVERSION)
 
 ### Includes and Defines (add further entries here):
 DEFINES	+= -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"' -DTNTVERSION=$(TNTVERSION) -DCXXTOOLVER=$(CXXTOOLVER)
 DEFINES	+= -DDISABLE_TEMPLATES_COLLIDING_WITH_STL
 VERSIONSUFFIX = gen_version_suffix.h
 
 ### The object files (add further files here):
 PLUGINOBJS := $(PLUGIN).o thread.o tntconfig.o setup.o i18n.o timers.o \
