diff -ru8bBwd a/projects/Amlogic-ce/devices/Amlogic-ng/packages/opengl-meson/package.mk b/projects/Amlogic-ce/devices/Amlogic-ng/packages/opengl-meson/package.mk
--- a/projects/Amlogic-ce/devices/Amlogic-ng/packages/opengl-meson/package.mk	2022-08-17 07:42:28.660348450 +0200
+++ b/projects/Amlogic-ce/devices/Amlogic-ng/packages/opengl-meson/package.mk	2022-08-14 12:03:58.655828884 +0200
@@ -41,16 +41,18 @@
   mkdir -p ${INSTALL}/usr/sbin
     cp ${PKG_DIR}/scripts/libmali-overlay-setup ${INSTALL}/usr/sbin
   # install needed files for compiling
   mkdir -p ${SYSROOT_PREFIX}/usr/include/EGL
     cp -pr include/EGL ${SYSROOT_PREFIX}/usr/include
     cp -pr include/EGL_platform/platform_fbdev/* ${SYSROOT_PREFIX}/usr/include/EGL
   mkdir -p ${SYSROOT_PREFIX}/usr/include/GLES2
     cp -pr include/GLES2 ${SYSROOT_PREFIX}/usr/include
+  mkdir -p ${SYSROOT_PREFIX}/usr/include/GLES3
+    cp -pr include/GLES3 ${SYSROOT_PREFIX}/usr/include
   mkdir -p ${SYSROOT_PREFIX}/usr/include/KHR
     cp -pr include/KHR ${SYSROOT_PREFIX}/usr/include
   mkdir -p ${SYSROOT_PREFIX}/usr/lib
     cp -pr lib/pkgconfig ${SYSROOT_PREFIX}/usr/lib
     ln ${INSTALL}/usr/lib/libMali.dvalin.so ${SYSROOT_PREFIX}/usr/lib/libEGL.so
     ln ${INSTALL}/usr/lib/libMali.dvalin.so ${SYSROOT_PREFIX}/usr/lib/libGLESv2.so
 }
