From a7480f91549f6e036f3f55e57c8f4b478a7e2ab1 Mon Sep 17 00:00:00 2001
From: Andreas Baierl <ichgeh@imkreisrum.de>
Date: Tue, 7 Jan 2025 13:09:46 +0100
Subject: [PATCH] quick ffmpeg 7 fix

---
 audio.c | 21 ++++++++++++---------
 1 file changed, 12 insertions(+), 9 deletions(-)

diff --git a/audio.c b/audio.c
index f20f164..8df00c7 100644
--- a/audio.c
+++ b/audio.c
@@ -513,10 +513,10 @@ static int AudioFilterInit(AVCodecContext *AudioCtx)
 
 	// Before filter init set HW parameter.
 	if (AudioCtx->sample_rate != (int)HwSampleRate ||
-		(AudioCtx->channels != (int)HwChannels && 
+		(AudioCtx->ch_layout.nb_channels != (int)HwChannels &&
 		!(AudioDownMix && HwChannels == 2))) {
 
-		err = AlsaSetup(AudioCtx->channels, AudioCtx->sample_rate, 0);
+		err = AlsaSetup(AudioCtx->ch_layout.nb_channels, AudioCtx->sample_rate, 0);
 		if (err)
 			return err;
 	}
@@ -535,10 +535,10 @@ static int AudioFilterInit(AVCodecContext *AudioCtx)
 		fprintf(stderr, "AudioFilterInit: Could not find the abuffer filter.\n");
 	if (!(abuffersrc_ctx = avfilter_graph_alloc_filter(filter_graph, abuffer, "src")))
 		fprintf(stderr, "AudioFilterInit: Could not allocate the abuffersrc_ctx instance.\n");
-	av_get_channel_layout_string(ch_layout, sizeof(ch_layout), AudioCtx->channels, AudioCtx->channel_layout);
+	av_channel_layout_describe(&AudioCtx->ch_layout, ch_layout, sizeof(ch_layout));
 #ifdef DEBUG
 	fprintf(stderr, "AudioFilterInit: IN ch_layout %s sample_fmt %s sample_rate %d channels %d\n",
-		ch_layout, av_get_sample_fmt_name(AudioCtx->sample_fmt), AudioCtx->sample_rate, AudioCtx->channels);
+		ch_layout, av_get_sample_fmt_name(AudioCtx->sample_fmt), AudioCtx->sample_rate, AudioCtx->ch_layout.nb_channels);
 #endif
 	av_opt_set    (abuffersrc_ctx, "channel_layout", ch_layout,                             AV_OPT_SEARCH_CHILDREN);
 	av_opt_set    (abuffersrc_ctx, "sample_fmt",     av_get_sample_fmt_name(AudioCtx->sample_fmt), AV_OPT_SEARCH_CHILDREN);
@@ -568,8 +568,11 @@ static int AudioFilterInit(AVCodecContext *AudioCtx)
 	}
 
 	// aformat
-	av_get_channel_layout_string(ch_layout, sizeof(ch_layout),
-		HwChannels, av_get_default_channel_layout(HwChannels));		// should use IN layout if more then 2 ch!?
+	AVChannelLayout channel_layout;
+	av_channel_layout_default(&channel_layout, HwChannels);
+	av_channel_layout_describe(&channel_layout, ch_layout, sizeof(ch_layout));
+	av_channel_layout_uninit(&channel_layout);
+	// should use IN layout if more then 2 ch!?
 #ifdef DEBUG
 	fprintf(stderr, "AudioFilterInit: OUT AudioDownMix %d HwChannels %d HwSampleRate %d ch_layout %s bytes_per_sample %d\n",
 		AudioDownMix, HwChannels, HwSampleRate,
@@ -1243,7 +1246,7 @@ void AudioEnqueue(AVFrame *frame)
 		return;
 	}
 
-	int count = frame->nb_samples * frame->channels * AudioBytesProSample;
+	int count = frame->nb_samples * frame->ch_layout.nb_channels * AudioBytesProSample;
 	buffer = (void *)frame->data[0];
 
 	if (AudioCompression) {		// in place operation
@@ -1253,7 +1256,7 @@ void AudioEnqueue(AVFrame *frame)
 		AudioNormalizer(buffer, count);
 	}
 
-	AudioReorderAudioFrame(buffer, count, frame->channels);
+	AudioReorderAudioFrame(buffer, count, frame->ch_layout.nb_channels);
 
 	pthread_mutex_lock(&AudioRbMutex);
 	n = RingBufferWrite(AudioRingBuffer, buffer, count);
@@ -1356,7 +1359,7 @@ in:
 			char errbuf[128];
 			av_strerror(err, errbuf, sizeof(errbuf));
 			fprintf(stderr, "AudioFilter: Error submitting the frame to the filter fmt %s channels %d %s\n",
-				av_get_sample_fmt_name(AudioCtx->sample_fmt), AudioCtx->channels, errbuf);
+				av_get_sample_fmt_name(AudioCtx->sample_fmt), AudioCtx->ch_layout.nb_channels, errbuf);
 			return;
 		} else {
 			Filterchanged = 1;
-- 
2.39.5

