diff -ru8bBwd a/video_drm.c b/video_drm.c
--- a/video_drm.c	2025-11-24 16:55:41.000000000 +0100
+++ b/video_drm.c	2026-08-16 13:34:22.969000000 +0200
@@ -1475,22 +1475,30 @@
 	av_free(par);
 
 	if (avfilter_graph_create_filter(&render->buffersink_ctx, buffersink, "out",
 		NULL, NULL, render->filter_graph) < 0)
 			fprintf(stderr, "VideoFilterInit: Cannot create buffer sink\n");
 
 	if (frame->format != AV_PIX_FMT_DRM_PRIME) {
 		enum AVPixelFormat pix_fmts[] = { AV_PIX_FMT_NV12, AV_PIX_FMT_NONE };
-		if (av_opt_set_int_list(render->buffersink_ctx, "pix_fmts", pix_fmts,
-				AV_PIX_FMT_NONE, AV_OPT_SEARCH_CHILDREN) < 0) {
+	    int ret;
+
+#if LIBAVFILTER_BUILD >= AV_VERSION_INT(10, 6, 100)
+	    ret = av_opt_set_array(render->buffersink_ctx, "pixel_formats",
+            AV_OPT_SEARCH_CHILDREN, 0, sizeof(pix_fmts) / sizeof(pix_fmts[0]) - 1, AV_OPT_TYPE_PIXEL_FMT, pix_fmts);
+#else
+	    ret = av_opt_set_int_list(render->buffersink_ctx, "pix_fmts",
+            pix_fmts, AV_PIX_FMT_NONE, AV_OPT_SEARCH_CHILDREN);
+#endif
+
+    if (ret < 0)
 			fprintf(stderr, "VideoFilterInit: Cannot set output pixel format\n");
 			goto fail;
 		}
-	}
 
 	outputs->name       = av_strdup("in");
 	outputs->filter_ctx = render->buffersrc_ctx;
 	outputs->pad_idx    = 0;
 	outputs->next       = NULL;
 
 	inputs->name       = av_strdup("out");
 	inputs->filter_ctx = render->buffersink_ctx;
