From 3c775effedde5a6651dbda6e721273884f2c087c Mon Sep 17 00:00:00 2001
From: Andreas Baierl <ichgeh@imkreisrum.de>
Date: Wed, 4 Oct 2023 09:39:20 +0200
Subject: [PATCH] fix debug

---
 requestresponse.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/requestresponse.cpp b/requestresponse.cpp
index c445c75..dc994dc 100644
--- a/requestresponse.cpp
+++ b/requestresponse.cpp
@@ -267,10 +267,10 @@ bool RequestResponse::Read(void* data_out, int bytes_to_read, int& bytes_read, C
 }
 
 void RequestResponse::GetResponseHeaders(CefRefPtr<CefResponse> response, int64 &response_length, CefString &redirectUrl) {
-    DEBUG("RequestResponse::GetResponseHeader: Error 1={}, Error2={}, Status1={}, Status2={}, StatusText={}",
-            url_request->GetResponse()->GetError(), url_request->GetRequestError(),
-            url_request->GetResponse()->GetStatus(), url_request->GetRequestStatus(),
-            url_request->GetResponse()->GetStatusText().ToString() );
+    DEBUG("RequestResponse::GetResponseHeader: Error1={}, Error2={}, Status1={}, Status2={}, StatusText={}",
+            (int)url_request->GetResponse()->GetError(), (int)url_request->GetRequestError(),
+            (int)url_request->GetResponse()->GetStatus(), (int)url_request->GetRequestStatus(),
+            url_request->GetResponse()->GetStatusText().ToString() );
 
     if (url_request->GetResponse()->GetStatus() != 200) {
         response->SetStatus(url_request->GetResponse()->GetStatus());
-- 
2.30.2

