diff --git a/kernel/vtuner.h b/kernel/vtuner.h
index 040feea..9dd8070 100644
--- a/kernel/vtuner.h
+++ b/kernel/vtuner.h
@@ -105,9 +105,9 @@ struct vtuner_signal
 
 #define VTUNER_MAJOR		226
 
-#define VTUNER_GET_MESSAGE	_IOR(VTUNER_MAJOR, 1, struct vtuner_message *)
-#define VTUNER_SET_RESPONSE 	_IOW(VTUNER_MAJOR, 2, struct vtuner_message *)
-#define VTUNER_SET_SIGNAL	_IOW(VTUNER_MAJOR, 3, struct vtuner_signal *)
+#define VTUNER_GET_MESSAGE	_IOR(VTUNER_MAJOR, 1, struct vtuner_message)
+#define VTUNER_SET_RESPONSE 	_IOW(VTUNER_MAJOR, 2, struct vtuner_message)
+#define VTUNER_SET_SIGNAL	_IOW(VTUNER_MAJOR, 3, struct vtuner_signal)
 
 #endif
 
diff --git a/kernel/vtunerc_ctrldev.c b/kernel/vtunerc_ctrldev.c
index 7d0a696..42346a0 100644
--- a/kernel/vtunerc_ctrldev.c
+++ b/kernel/vtunerc_ctrldev.c
@@ -204,7 +204,7 @@ static int vtunerc_ctrldev_close(struct inode *inode, struct file *filp)
 		ctx->stat_wr_data = 0;
 		ctx->stat_fi_data = 0;
 		ctx->stat_fe_data = 0;
-		memset(&ctx->signal.status,0,sizeof(struct vtuner_signal));
+		memset(&ctx->signal,0,sizeof(struct vtuner_signal));
 		ctx->fe_params.delivery_system=0; // now retune can happen
 		memset(&ctx->pusitab,0,sizeof(unsigned char)*MAX_PIDTAB_LEN);
 	}
@@ -263,6 +263,10 @@ static long vtunerc_ctrldev_ioctl(struct file *file, unsigned int cmd, unsigned
 	return ret;
 }
 
+static long compat_vtunerc_ctrldev_ioctl(struct file *file, unsigned int cmd, unsigned long arg) {
+    return vtunerc_ctrldev_ioctl(file, cmd, arg);
+}
+
 static unsigned int vtunerc_ctrldev_poll(struct file *filp, poll_table *wait)
 {
 	struct vtunerc_ctx *ctx = filp->private_data;
@@ -282,6 +286,7 @@ static unsigned int vtunerc_ctrldev_poll(struct file *filp, poll_table *wait)
 static const struct file_operations vtunerc_ctrldev_fops = {
 	.owner = THIS_MODULE,
 	.unlocked_ioctl = vtunerc_ctrldev_ioctl,
+	.compat_ioctl = compat_vtunerc_ctrldev_ioctl,
 	.write = vtunerc_ctrldev_write,
 	.read  = vtunerc_ctrldev_read,
 	.poll  = (void *) vtunerc_ctrldev_poll,
diff --git a/satip/log.c b/satip/log.c
index 1454500..89f653d 100644
--- a/satip/log.c
+++ b/satip/log.c
@@ -45,7 +45,7 @@ void write_message(const unsigned int mtype, const int level, const char* fmt, .
     va_start(ap, fmt);
     vsnprintf(tn, sizeof(tn), fmt, ap);
     va_end(ap);
-    strncat(msg, tn, sizeof(msg));
+    strncat(msg, tn, sizeof(msg)-1);
 
     if(use_syslog) {
       int priority;
@@ -82,7 +82,7 @@ void append_message(const char* fmt, ... ) {
   vsnprintf(tn, sizeof(tn), fmt, ap);
   va_end(ap);
 
-  strncat(msg, tn, sizeof(msg));
+  strncat(msg, tn, sizeof(msg)-1);
 }  
 
 int open_udplog(char *ipaddr, int portnum) {
