diff -ru8abBw a/kernel/vtunerc_priv.h b/kernel/vtunerc_priv.h
--- a/kernel/vtunerc_priv.h	2024-01-03 12:49:32.000000000 +0100
+++ b/kernel/vtunerc_priv.h	2024-01-07 15:29:09.733115166 +0100
@@ -67,17 +67,17 @@
 	struct semaphore xchange_sem;
 	struct semaphore ioctl_sem;
 	struct semaphore tswrite_sem;
 	int fd_opened;
 	int adapter_inuse;
 
 	char *procname;
 
-	char *kernel_buf;
+	unsigned char *kernel_buf;
 	ssize_t kernel_buf_size;
 
 	/* ctrldev */
 	int noresponse;
 	struct vtuner_message ctrldev_request;
 	struct vtuner_message ctrldev_response;
 	wait_queue_head_t ctrldev_wait_request_wq;
 	wait_queue_head_t ctrldev_wait_response_wq;
