diff -ru8bBw a/Makefile b/Makefile
--- a/Makefile	2023-04-04 18:09:22.000000000 +0200
+++ b/Makefile	2023-04-08 07:51:53.511716520 +0200
@@ -77,17 +77,17 @@
 $(SOFILE): $(OBJS)
 	$(CXX) $(CXXFLAGS) $(LDFLAGS) -rdynamic -shared $(OBJS) $(LIBS) -o $@
 
 plugins:
 	@find $(PLGSRCDIR) -maxdepth 1 -type d -name "[a-z0-9]*" -exec \
       $(MAKE) \-\-no-print-directory -C {} \;
 
 install-lib: $(SOFILE)
-	install -D $^ $(DESTDIR)$(LIBDIR)/$^.$(APIVERSION)
+	install -D $^ $(LIBDIR)/$^.$(APIVERSION)
 
 install-conf:
 	@mkdir -p $(DESTDIR)$(PLGCONFDIR)
 	@if [ ! -f $(DESTDIR)$(PLGCONFDIR)/override.conf ]; then cp conf/override.conf $(DESTDIR)$(PLGCONFDIR); fi;
 
 install-plugins: plugins
 	mkdir -p "$(_PLGDEST)"
 	for i in ${PLGSRCDIR}/*/Makefile; do\
