diff --git a/packages/addons/addon-depends/tntnet/package.mk b/packages/addons/addon-depends/tntnet/package.mk
index c783c93df8..3e96385dfd 100644
--- a/packages/addons/addon-depends/tntnet/package.mk
+++ b/packages/addons/addon-depends/tntnet/package.mk
@@ -7,7 +7,7 @@ PKG_VERSION="2.2.1"
 PKG_SHA256="c83170d08ef04c5868051e1c28c74b9562fe71e9e8263828e755ad5bd3547521"
 PKG_LICENSE="GPL-2"
 PKG_SITE="http://www.tntnet.org/"
-PKG_URL="http://www.tntnet.org/download/${PKG_NAME}-${PKG_VERSION}.tar.gz"
+PKG_URL="http://ftp.rz.tu-braunschweig.de/pub/mirror/ubuntu-packages/pool/universe/t/${PKG_NAME}/${PKG_NAME}_${PKG_VERSION}.orig.tar.gz"
 PKG_DEPENDS_HOST="cxxtools:host zlib:host"
 PKG_DEPENDS_TARGET="toolchain tntnet:host libtool cxxtools zlib"
 PKG_LONGDESC="A web application server for C++."
