diff --git a/packages/addons/addon-depends/cxxtools/package.mk b/packages/addons/addon-depends/cxxtools/package.mk
index 7fe56f9ebc..231f491ea9 100644
--- a/packages/addons/addon-depends/cxxtools/package.mk
+++ b/packages/addons/addon-depends/cxxtools/package.mk
@@ -7,7 +7,7 @@ PKG_VERSION="2.2.1"
 PKG_SHA256="8cebb6d6cda7c93cc4f7c0d552a68d50dd5530b699cf87916bb3b708fdc4e342"
 PKG_LICENSE="GPL-2"
 PKG_SITE="http://www.tntnet.org/cxxtools.html"
-PKG_URL="http://www.tntnet.org/download/${PKG_NAME}-${PKG_VERSION}.tar.gz"
+PKG_URL="http://ftp.rz.tu-braunschweig.de/pub/mirror/ubuntu-packages/pool/universe/c/${PKG_NAME}/${PKG_NAME}_${PKG_VERSION}.orig.tar.gz"
 PKG_DEPENDS_HOST="toolchain:host"
 PKG_DEPENDS_TARGET="toolchain cxxtools:host"
 PKG_LONGDESC="Cxxtools is a collection of general-purpose C++ classes."
