diff --git a/meson.build b/meson.build
index 9da2047..6fdbf26 100644
--- a/meson.build
+++ b/meson.build
@@ -71,22 +56,19 @@ remotrans = executable('remotrans', 'main.cpp', 'logger.cpp', 'streamhandler.cpp
                         include_directories : incdir,
                         cpp_args : trans_compile_args,
                         install : true,
-                        install_dir : meson.current_build_dir() + '/Release',
                         dependencies: [mini_dep, spdlog_dep, tiny_process_library_dep, openssl_dep, thrift_dep])
 
 r_dash2ts = executable('r_dash2ts', 'r_dash2ts.cpp', 'logger.cpp',
                        cpp_args : trans_compile_args,
                        install : true,
-                       install_dir : meson.current_build_dir() + '/Release',
                        dependencies: [spdlog_dep, dash2ts_dep])
 
 r_dash2tsd = executable('r_dash2ts_debug', 'r_dash2ts_debug.cpp', 'logger.cpp',
                        cpp_args : trans_compile_args,
                        install : true,
-                       install_dir : meson.current_build_dir() + '/Release',
                        dependencies: [spdlog_dep, dash2ts_dep])
 
 #
 # install
 #
-install_subdir('static-content/movie', install_dir : meson.current_build_dir() + '/Release')
+install_subdir('static-content/movie', install_dir : '')
