diff --git a/meson.build b/meson.build
index 076f990..c7ac2d8 100644
--- a/meson.build
+++ b/meson.build
@@ -30,10 +30,9 @@ mini_dep = mini_proj.get_variable('mini_dep')
 
 exe = executable('remotrans', 'main.cpp', 'logger.cpp', 'ffmpeghandler.cpp', 'browserclient.cpp', 'transcodeconfig.cpp', 'vdrclient.cpp',
                  install : true,
-                 install_dir : meson.current_build_dir() + '/Release',
                  dependencies: [mini_dep, spdlog_dep, tiny_process_library_dep])
 
 #
 # install
 #
-install_subdir('static-content/movie', install_dir : meson.current_build_dir() + '/Release')
+install_subdir('static-content/movie', install_dir : '')
