diff -ru8bBwd a/configure.ac b/configure.ac
--- a/configure.ac	2024-06-05 17:56:07.000000000 +0200
+++ b/configure.ac	2024-08-10 10:27:41.507091457 +0200
@@ -3065,46 +3065,20 @@
 dnl  Video plugins
 dnl
 
 EXTEND_HELP_STRING([Video plugins:])
 
 dnl
 dnl  OpenGL
 dnl
-PKG_CHECK_MODULES([EGL], [egl], [
-  have_egl="yes"
-], [
   have_egl="no"
-])
 AM_CONDITIONAL([HAVE_EGL], [test "${have_egl}" = "yes"])
 
 have_gl="no"
-PKG_CHECK_MODULES([GL], [gl], [
-  have_gl="yes"
-], [
-  AC_MSG_CHECKING([for OpenGL])
-  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
-#ifdef _WIN32
-# include <GL/glew.h>
-#endif
-#include <GL/gl.h>
-]], [
-    [int t0 = GL_TEXTURE0;]])
-  ], [
-    GL_CFLAGS=""
-    have_gl="yes"
-    AS_IF([test "${SYS}" != "mingw32"], [
-      GL_LIBS="-lGL"
-    ], [
-      GL_LIBS="-lopengl32"
-    ])
-  ])
-  AC_MSG_RESULT([${have_gl}])
-])
 AM_CONDITIONAL([HAVE_GL], [test "${have_gl}" = "yes"])
 AS_IF([test "${have_gl}" = "yes"], [
   AC_DEFINE([HAVE_GL], 1, [Defined if having OpenGL])
 ])
 
 dnl OpenGL ES 2: depends on EGL 1.1
 PKG_ENABLE_MODULES_VLC([GLES2], [], [glesv2], [OpenGL ES v2 support], [disabled])
 
