diff -ru8bBwd a/modules/misc/gnutls.c b/modules/misc/gnutls.c
--- a/modules/misc/gnutls.c	2024-06-05 17:56:07.000000000 +0200
+++ b/modules/misc/gnutls.c	2024-08-10 12:55:34.548848694 +0200
@@ -784,17 +784,17 @@
     set_shortname( "GNU TLS" )
     set_description( N_("GNU TLS transport layer security") )
     set_capability( "tls client", 1 )
     set_callbacks( OpenClient, CloseClient )
     set_category( CAT_ADVANCED )
     set_subcategory( SUBCAT_ADVANCED_NETWORK )
     add_bool("gnutls-system-trust", true, SYSTEM_TRUST_TEXT,
              SYSTEM_TRUST_LONGTEXT, true)
-    add_string("gnutls-dir-trust", NULL, DIR_TRUST_TEXT,
+    add_string("gnutls-dir-trust", "/etc/pki/tls/certs/", DIR_TRUST_TEXT,
                DIR_TRUST_LONGTEXT, true)
     add_string ("gnutls-priorities", "NORMAL", PRIORITIES_TEXT,
                 PRIORITIES_LONGTEXT, false)
         change_string_list (priorities_values, priorities_text)
 #ifdef ENABLE_SOUT
     add_submodule ()
         set_description( N_("GNU TLS server") )
         set_capability( "tls server", 1 )
