diff --git a/distributions/LibreELEC/options b/distributions/LibreELEC/options
index 5a663f4f05..d9d081a9ec 100644
--- a/distributions/LibreELEC/options
+++ b/distributions/LibreELEC/options
@@ -67,7 +67,7 @@
   ADDITIONAL_DRIVERS="RTL8192DU RTL8812AU"
 
 # Default size of system partition, in MB, eg. 512
-  SYSTEM_SIZE=512
+  SYSTEM_SIZE=1024
 
 # Default system partition offset, in sectors, eg. 2048
   SYSTEM_PART_START=8192
