diff -ru8bBwd a/player-dvd.c b/player-dvd.c
--- a/player-dvd.c	2026-09-14 17:55:31.049640095 +0200
+++ b/player-dvd.c	2026-09-14 17:57:19.700000000 +0200
@@ -3102,29 +3102,29 @@
 
     if (!DVDActiveAndRunning()) {
         *AudioLanguageStr = "n.a.";
         return;
     }
 
     switch (currentNavAudioTrackType) {
         case aAC3:
-            audioTypeDescr = "ac3";
+            audioTypeDescr = const_cast<char*>("ac3");
             break;
         case aDTS:
-            audioTypeDescr = "dts";
+            audioTypeDescr = const_cast<char*>("dts");
             break;
         case aLPCM:
-            audioTypeDescr = "pcm";
+            audioTypeDescr = const_cast<char*>("pcm");
             break;
         case aMPEG:
-            audioTypeDescr = "mp2";
+            audioTypeDescr = const_cast<char*>("mp2");
             break;
         default:
-            audioTypeDescr = "non";
+            audioTypeDescr = const_cast<char*>("non");
             break;
     }
 
     uint16_t audioStreamLanguageCode = GetAudioTrackLanguageCode(currentNavAudioTrack);
     char audioLanguageStr[3] = {(char)audioStreamLanguageCode, (char)(audioStreamLanguageCode >> 8), 0};
     if (GetAudioStreamNumbers() > 1)
         sprintf(buffer,"%s %d/%d %s", audioStreamLanguageCode != 0xFFFF ? audioLanguageStr : "", GetCurrentNavAudioTrackIdx() + 1, GetAudioStreamNumbers() - 1, audioTypeDescr);
     else
