diff -ru8bBw a/Makefile b/Makefile
--- a/Makefile	2024-04-28 10:56:26.000000000 +0200
+++ b/Makefile	2024-05-20 10:26:45.211807392 +0200
@@ -37,17 +37,17 @@
 CONFIG += -DAV_INFO -DAV_INFO_TIME=3000	# info/debug a/v sync
 CONFIG += -DUSE_MPEG_COMPLETE		# support only complete mpeg packets
 CONFIG += -DUSE_VDR_SPU			# use VDR SPU decoder.
 CONFIG += -DUSE_OPENGLOSD 
 
 ### The version number of this plugin (taken from the main source file):
 
 VERSION = $(shell grep 'static const char \*const VERSION *=' softhdodroid.cpp | awk '{ print $$7 }' | sed -e 's/[";]//g')
-GIT_REV = $(shell git describe --always 2>/dev/null)
+GIT_REV ?= $(shell git describe --always 2>/dev/null)
 ### The name of the distribution archive:
 
 ### The directory environment:
 
 # Use package data if installed...otherwise assume we're under the VDR source directory:
 PKGCFG = $(if $(VDRDIR),$(shell pkg-config --variable=$(1) $(VDRDIR)/vdr.pc),$(shell PKG_CONFIG_PATH="$$PKG_CONFIG_PATH:../../.." pkg-config --variable=$(1) vdr))
 LIBDIR = $(call PKGCFG,libdir)
 LOCDIR = $(call PKGCFG,locdir)
