diff --git a/packages/addons/addon-depends/system-tools-depends/libgpiod/package.mk b/packages/addons/addon-depends/system-tools-depends/libgpiod/package.mk
index cd61e40988..9cb4f51e18 100644
--- a/packages/addons/addon-depends/system-tools-depends/libgpiod/package.mk
+++ b/packages/addons/addon-depends/system-tools-depends/libgpiod/package.mk
@@ -3,10 +3,10 @@
 
 PKG_NAME="libgpiod"
 PKG_VERSION="1.6.4"
-PKG_SHA256="829d4ac268df07853609d67cfc7f476e9aa736cb2a68a630be99e8fad197be0a"
+PKG_SHA256="7b146e12f28fbca3df7557f176eb778c5ccf952ca464698dba8a61b2e1e3f9b5"
 PKG_LICENSE="GPLv2+"
 PKG_SITE="https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git/about/"
-PKG_URL="https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git/snapshot/libgpiod-${PKG_VERSION}.tar.gz"
+PKG_URL="https://mirrors.edge.kernel.org/pub/software/libs/libgpiod/libgpiod-${PKG_VERSION}.tar.xz"
 PKG_DEPENDS_TARGET="toolchain"
 PKG_LONGDESC="Tools for interacting with the linux GPIO character device."
 PKG_TOOLCHAIN="autotools"
