diff --git a/packages/sysutils/keyutils/package.mk b/packages/sysutils/keyutils/package.mk
index a1d9ddd198..d3bf0c6d6d 100644
--- a/packages/sysutils/keyutils/package.mk
+++ b/packages/sysutils/keyutils/package.mk
@@ -7,7 +7,7 @@ PKG_VERSION="1.6.3"
 PKG_SHA256="a61d5706136ae4c05bd48f86186bcfdbd88dd8bd5107e3e195c924cfc1b39bb4"
 PKG_LICENSE="GPL"
 PKG_SITE="https://git.kernel.org/pub/scm/linux/kernel/git/dhowells/keyutils.git/"
-PKG_URL="https://git.kernel.org/pub/scm/linux/kernel/git/dhowells/keyutils.git/snapshot/${PKG_NAME}-${PKG_VERSION}.tar.gz"
+PKG_URL="http://deb.debian.org/debian/pool/main/k/keyutils/keyutils_${PKG_VERSION}.orig.tar.gz"
 PKG_DEPENDS_TARGET="toolchain"
 PKG_LONGDESC="Keyutils is a set of utilities for managing the key retention facility in the kernel."
 PKG_BUILD_FLAGS="+pic"
