diff -ru8bBw a/Makefile b/Makefile
--- a/Makefile	2020-07-03 14:36:36.000000000 +0200
+++ b/Makefile	2022-04-17 16:41:08.144011004 +0200
@@ -47,17 +47,17 @@
 ### The name of the distribution archive:
 
 ARCHIVE = $(PLUGIN)-$(VERSION)
 PACKAGE = vdr-$(ARCHIVE)
 
 ### The name of the shared object file:
 
 SOFILE = libvdr-$(PLUGIN).so
-CMDFILE = $(PLUGIN)ctl-$(APIVERSION)
+CMDFILE = $(PLUGIN)ctl
 
 
 ### Includes and Defines (add further entries here):
 
 INCLUDES += -I.
 
 DEFINES += -DPLUGIN_NAME_I18N='"$(PLUGIN)"' -DAPIVERSION='"$(APIVERSION)"'
 #DEFINES += -DDEBUG
@@ -125,7 +125,7 @@ install-lib: $(SOFILE)
 	install -D $^ $(DESTDIR)$(LIBDIR)/$^.$(APIVERSION)
 
 install-cmd: cmd
-	cp $(CMDFILE) /usr/local/bin
+	install -D $(CMDFILE) $(DESTDIR)/usr/local/bin/$(CMDFILE)
 
 install: install-lib install-i18n install-cmd
 
