diff -ru8bBwd a/gprofng/src/Makefile.am b/gprofng/src/Makefile.am
--- a/gprofng/src/Makefile.am	2025-02-02 01:00:00.000000000 +0100
+++ b/gprofng/src/Makefile.am	2025-10-16 09:45:17.190257918 +0200
@@ -174,15 +174,15 @@
 # so ensure that the necessary libraries are built at dist time.
 dist-hook: $(LIBGPROFNG)
 
 $(srcdir)/DbeSession.cc: QLParser.tab.hh
 
 .PHONY: install-exec-local
 install-exec-local:
 	$(mkinstalldirs) $(DESTDIR)$(bindir)
-	rm -f $(DESTDIR)$(bindir)/gp-{archive,collect-app,display-html,display-src,display-text}
+	rm -f $(DESTDIR)$(bindir)/gp-archive $(DESTDIR)$(bindir)/gp-collect-app $(DESTDIR)$(bindir)/gp-display-html $(DESTDIR)$(bindir)/gp-display-src $(DESTDIR)$(bindir)/gp-display-text
 	ln -s gprofng-archive $(DESTDIR)$(bindir)/gp-archive
 	ln -s gprofng-collect-app $(DESTDIR)$(bindir)/gp-collect-app
 	ln -s gprofng-display-html $(DESTDIR)$(bindir)/gp-display-html
 	ln -s gprofng-display-src $(DESTDIR)$(bindir)/gp-display-src
 	ln -s gprofng-display-text $(DESTDIR)$(bindir)/gp-display-text
 
diff -ru8bBwd a/gprofng/src/Makefile.in b/gprofng/src/Makefile.in
--- a/gprofng/src/Makefile.in	2025-02-02 01:00:00.000000000 +0100
+++ b/gprofng/src/Makefile.in	2025-10-16 10:24:41.205472303 +0200
@@ -1114,17 +1114,17 @@
 # so ensure that the necessary libraries are built at dist time.
 dist-hook: $(LIBGPROFNG)
 
 $(srcdir)/DbeSession.cc: QLParser.tab.hh
 
 .PHONY: install-exec-local
 install-exec-local:
 	$(mkinstalldirs) $(DESTDIR)$(bindir)
-	rm -f $(DESTDIR)$(bindir)/gp-{archive,collect-app,display-html,display-src,display-text}
+	rm -f $(DESTDIR)$(bindir)/gp-archive $(DESTDIR)$(bindir)/gp-collect-app $(DESTDIR)$(bindir)/gp-display-html $(DESTDIR)$(bindir)/gp-display-src $(DESTDIR)$(bindir)/gp-display-text
 	ln -s gprofng-archive $(DESTDIR)$(bindir)/gp-archive
 	ln -s gprofng-collect-app $(DESTDIR)$(bindir)/gp-collect-app
 	ln -s gprofng-display-html $(DESTDIR)$(bindir)/gp-display-html
 	ln -s gprofng-display-src $(DESTDIR)$(bindir)/gp-display-src
 	ln -s gprofng-display-text $(DESTDIR)$(bindir)/gp-display-text
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
