diff -ru8bBwd a/gprofng/src/Makefile.am b/gprofng/src/Makefile.am
--- a/gprofng/src/Makefile.am	2025-11-10 01:00:00.000000000 +0100
+++ b/gprofng/src/Makefile.am	2025-11-14 06:31:55.642881696 +0100
@@ -175,13 +175,13 @@
 # so ensure that the necessary libraries are built at dist time.
 dist-hook: $(LIBGPROFNG)
 
 $(srcdir)/DbeSession.cc: QLParser.tab.hh
 
 .PHONY: install-exec-local
 install-exec-local:
 	$(mkinstalldirs) $(DESTDIR)$(bindir)
-	for i in gp-{archive,collect-app,display-html,display-src,display-text}; do \
+	for i in gp-archive gp-collect-app gp-display-html gp-display-src gp-display-text; do \
 		oldname=`echo $$i | sed '$(transform)'`; \
 		rm -f $(DESTDIR)$(bindir)/$$oldname ; \
 		ln -s `echo $$oldname | sed 's&gp-&gprofng-&'` $(DESTDIR)$(bindir)/$$oldname; \
 	done
diff -ru8bBwd a/gprofng/src/Makefile.in b/gprofng/src/Makefile.in
--- a/gprofng/src/Makefile.in	2025-11-10 01:00:00.000000000 +0100
+++ b/gprofng/src/Makefile.in	2025-11-14 06:33:09.519471549 +0100
@@ -1116,17 +1116,17 @@
 # so ensure that the necessary libraries are built at dist time.
 dist-hook: $(LIBGPROFNG)
 
 $(srcdir)/DbeSession.cc: QLParser.tab.hh
 
 .PHONY: install-exec-local
 install-exec-local:
 	$(mkinstalldirs) $(DESTDIR)$(bindir)
-	for i in gp-{archive,collect-app,display-html,display-src,display-text}; do \
+	for i in gp-archive gp-collect-app gp-display-html gp-display-src gp-display-text; do \
 		oldname=`echo $$i | sed '$(transform)'`; \
 		rm -f $(DESTDIR)$(bindir)/$$oldname ; \
 		ln -s `echo $$oldname | sed 's&gp-&gprofng-&'` $(DESTDIR)$(bindir)/$$oldname; \
 	done
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
