diff -ru8bBw a/softhdodroid.cpp b/softhdodroid.cpp
--- a/softhdodroid.cpp	2023-07-24 15:38:26.000000000 +0200
+++ b/softhdodroid.cpp	2023-07-25 15:10:30.154812387 +0200
@@ -639,16 +639,17 @@
 class cDummyPixmap : public cPixmap {
 public:
     cDummyPixmap(int Layer, const cRect &ViewPort, const cRect &DrawPort = cRect::Null) : cPixmap(Layer, ViewPort, DrawPort) {}
     virtual ~cDummyPixmap(void) {}
     virtual void Clear(void) {}
     virtual void Fill(tColor Color) { (void)Color; }
     virtual void DrawImage(const cPoint &Point, const cImage &Image) { (void)Point; (void)Image; }
     virtual void DrawImage(const cPoint &Point, int ImageHandle) { (void)Point; (void)ImageHandle; }
+    virtual void DrawImageScaled(const cPoint &Point, const cImage &Image, double FactorX, double FactorY) { (void)Point; (void)Image; (void)FactorX; (void)FactorY; };
     virtual void DrawPixel(const cPoint &Point, tColor Color) { (void)Point; (void)Color; }
     virtual void DrawBitmap(const cPoint &Point, const cBitmap &Bitmap, tColor ColorFg = 0, tColor ColorBg = 0, bool Overlay = false) {
         (void) Point; (void)Bitmap; (void)ColorFg; (void)ColorBg; (void)Overlay; }
     virtual void DrawText(const cPoint &Point, const char *s, tColor ColorFg, tColor ColorBg, const cFont *Font, int Width = 0, int Height = 0, int Alignment = taDefault) {
         (void)Point; (void)s; (void)ColorFg; (void) ColorBg; (void) Font; (void)Width; (void)Height; (void)Alignment; }
     virtual void DrawRectangle(const cRect &Rect, tColor Color) { (void)Rect; (void)Color; }
     virtual void DrawEllipse(const cRect &Rect, tColor Color, int Quadrants = 0) { (void)Rect; (void)Color; (void)Quadrants; }
     virtual void DrawSlope(const cRect &Rect, tColor Color, int Type) { (void)Rect; (void)Color; (void)Type; }
