diff -ru8bBw a/codec.c b/codec.c
--- a/codec.c	2024-06-17 15:18:38.000000000 +0200
+++ b/codec.c	2024-07-15 19:32:13.037434924 +0200
@@ -538,26 +538,29 @@
     if ((err = snd_mixer_load(handle)) < 0)
     {
       Debug(3, "CAESinkALSA- Mixer %s load error: %s", card, snd_strerror(err));
       snd_mixer_close(handle);
       return;
     }
     
     if (myKernel == 5) {
+        /*
         Debug(3,"CAESinkALSA - Set HDMITX Source to spdif_b \n");
         snd_mixer_selem_id_set_name(sid, "HDMITX Audio Source Select");
         elem = snd_mixer_find_selem(handle, sid);
         if (!elem) {
             Debug(3, "CAESinkALSA - Unable to find simple control '%s',%i\n",
                 snd_mixer_selem_id_get_name(sid), snd_mixer_selem_id_get_index(sid));
             snd_mixer_close(handle);
             return;
         } else
         snd_mixer_selem_set_enum_item(elem, (snd_mixer_selem_channel_id_t)0, 1); // 0 = spdif  1= spdif_b
+        */
+
         Debug(3,"CAESinkALSA - Set SPDIF CLK Fine Setting \n");
         snd_mixer_selem_id_set_name(sid, "SPDIF CLK Fine Setting");
         elem = snd_mixer_find_selem(handle, sid);
         if (!elem) {
             Debug(3, "CAESinkALSA - Unable to find simple control '%s',%i\n",
                 snd_mixer_selem_id_get_name(sid), snd_mixer_selem_id_get_index(sid));
             snd_mixer_close(handle);
             return;
