diff -ru8bBwd a/softhddev.c b/softhddev.c
--- a/softhddev.c	2026-09-25 15:56:35.000000000 +0200
+++ b/softhddev.c	2026-09-27 11:12:22.191000000 +0200
@@ -2780,30 +2780,28 @@
         if (timeout) {                  // let display thread work
             usleep(timeout * 1000);
         }
         return !atomic_read(&MyVideoStream->PacketsFilled);
     }
     return 1;
 }
 
-#if APIVERSNUM >= 30014
 extern int amlGetBuffer(int pip, int filled);
-int Drain()
+int SoftDrainDevice()
 {
    int fill_percent;
     fill_percent = amlGetBuffer(MyVideoStream->HwDecoder->pip, 1);
     Debug(3, "Buffer Fill in Percent: %d\n", fill_percent);
     if (fill_percent == 0) {
         return 1; //replay can be ended
     } else {
         return 0; ////integer makes bool cSoftHdDevice::Drain false
     }
 }
-#endif
 
 //////////////////////////////////////////////////////////////////////////////
 //  OSD
 //////////////////////////////////////////////////////////////////////////////
 
 /**
 **  Get OSD size and aspect.
 **
diff -ru8bBwd a/softhddev.h b/softhddev.h
--- a/softhddev.h	2026-09-25 15:56:35.000000000 +0200
+++ b/softhddev.h	2026-09-27 11:05:35.752000000 +0200
@@ -70,17 +70,17 @@
     /// C plugin display I-frame as a still picture.
     extern void StillPicture(const uint8_t *, int);
     /// C plugin poll if ready
     extern int Poll(int);
     /// C plugin flush output buffers
     extern int Flush(int);
 #if APIVERSNUM >= 30014
     /// C plugin drain output buffers
-    extern int Drain();
+    extern int SoftDrainDevice();
 #endif
 
     /// C plugin command line help
     extern const char *CommandLineHelp(void);
     /// C plugin process the command line arguments
     extern int ProcessArgs(int, char *const[]);
 
     /// C plugin exit + cleanup
diff -ru8bBwd a/softhdodroid.cpp b/softhdodroid.cpp
--- a/softhdodroid.cpp	2026-09-25 15:56:35.000000000 +0200
+++ b/softhdodroid.cpp	2026-09-27 11:12:31.295000000 +0200
@@ -2243,17 +2243,17 @@
     virtual void Clear(void);
     virtual void Play(void);
     virtual void Freeze(void);
     virtual void Mute(void);
     virtual void StillPicture(const uchar *, int);
     virtual bool Poll(cPoller &, int = 0);
     virtual bool Flush(int = 0);
 #if APIVERSNUM >= 30014
-    virtual bool Drain(void);
+    virtual bool DrainDevice(void);
 #endif
     virtual int64_t GetSTC(void);
     virtual cRect CanScaleVideo(const cRect &, int = taCenter);
     virtual void ScaleVideo(const cRect & = cRect::Null);
     virtual void SetVideoDisplayFormat(eVideoDisplayFormat);
     virtual void SetVideoFormat(bool);
     virtual void GetVideoSize(int &, int &, double &);
     virtual void GetOsdSize(int &, int &, double &);
@@ -2534,21 +2534,20 @@
 bool cSoftHdDevice::Flush(int timeout_ms)
 {
     dsyslog("[softhddev]%s: %d ms\n", __FUNCTION__, timeout_ms);
 
     return::Flush(timeout_ms);
 }
 
 #if APIVERSNUM >= 30014
-bool cSoftHdDevice::Drain(void)
+bool cSoftHdDevice::DrainDevice(void)
 {
-    dsyslog("[softhddev]%s:\n", __FUNCTION__);
-
-    return::Drain();
+    // dsyslog("[softhddev]%s:\n", __FUNCTION__);
+    return SoftDrainDevice() ? true : false;
 }
 #endif
 
 // ----------------------------------------------------------------------------
 
 /**
 **  Sets the video display format to the given one (only useful if this
 **  device has an MPEG decoder).
