diff -ru8Bbwd a/Makefile b/Makefile
--- a/Makefile	2026-09-25 15:56:35.000000000 +0200
+++ b/Makefile	2026-09-25 21:04:40.992000000 +0200
@@ -63,16 +63,19 @@
 
 ifeq ($(CFLAGS),)
 $(warning CFLAGS not set)
 endif
 ifeq ($(CXXFLAGS),)
 $(warning CXXFLAGS not set)
 endif
 
+CFLAGS += $(EXTRAFLAGS)
+CXXFLAGS += $(EXTRAFLAGS)
+
 ### The version number of VDR's plugin API:
 
 APIVERSION = $(call PKGCFG,apiversion)
 
 ### Allow user defined options to overwrite defaults:
 
 -include $(PLGCFG)
 
diff -ru8Bbwd a/openglosd.cpp b/openglosd.cpp
--- a/openglosd.cpp	2026-09-25 15:56:35.000000000 +0200
+++ b/openglosd.cpp	2026-09-25 21:07:05.068000000 +0200
@@ -27,17 +27,17 @@
 
 //#include <GL/glew.h>
 #define EGL_EGLEXT_PROTOTYPES
 #define MESA_EGL_NO_X11_HEADERS
 #include <EGL/egl.h>
 #include <EGL/eglplatform.h>
 #include <libdrm/drm_fourcc.h>
 #include <EGL/eglext.h>
-#ifndef _MALI_FBDEV_TYPES_H_ 
+#ifdef _CE22_
 #include <gbm.h>
 #endif
 #include <xf86drm.h>
 #include <xf86drmMode.h>
 
 #include "ge2d.h"
 #include "ge2d_cmd.h"
 #include "ion.h"
@@ -59,24 +59,23 @@
 #define EGL_DMA_BUF_PLANE1_PITCH_EXT    0x3277
 #define EGL_DMA_BUF_PLANE2_FD_EXT       0x3278
 #define EGL_DMA_BUF_PLANE2_OFFSET_EXT   0x3279
 #define EGL_DMA_BUF_PLANE2_PITCH_EXT    0x327A
 #define EGL_YUV_COLOR_SPACE_HINT_EXT    0x327B
 
 #define USE_DRM
 
-
 //int fd = -1;
 int ge2d_fd = -1;
 int ion_fd = -1;
 
 
 int drm_fd = -1;
-#ifndef _MALI_FBDEV_TYPES_H_ 
+#ifdef _CE22_
 static struct gbm_device *gbm_dev;
 #endif
 EGLContext egl_ctx;
 EGLDisplay egl_dpy;
 
 
 int eglInitiated = 0;
 
@@ -1948,17 +1947,17 @@
     for (int i = 0; i < OGL_MAX_OSDIMAGES; i++) {
         if (imageCache[i].used) {
             DropImageData(i);
         }
     }
     Cancel(2);
     stalled = false;
 
-#ifndef _MALI_FBDEV_TYPES_H_ 
+#ifdef _CE22_
     if (drm_fd >= 0) {
         gbm_device_destroy(gbm_dev);
         close(drm_fd);
     }
     drm_fd = -1;
 #endif
 }
 
@@ -2174,17 +2173,17 @@
 
     egl_dpy = eglGetDisplay(nativeDisplay);
     
     // Initialize EGL
     EGLint major;
     EGLint minor;
     EGLBoolean success;
 
-#ifdef _MALI_FBDEV_TYPES_H_ 
+#ifndef _CE22_
     success = eglInitialize( egl_dpy, &major, &minor);
     if (success != EGL_TRUE)
     {
         printf("Unable to open Display\n");
         return false;
     }
 #else
     printf("Try Mesa GBM\n");
@@ -2214,17 +2213,22 @@
     EGLConfig *configs = nullptr;
     EGLint num_configs;
     EGLint attribs[] = {
         EGL_SURFACE_TYPE, EGL_WINDOW_BIT,
         EGL_RED_SIZE, 8, 
         EGL_GREEN_SIZE, 8, 
         EGL_BLUE_SIZE, 8, 
         EGL_ALPHA_SIZE,6, 
-        EGL_RENDERABLE_TYPE, EGL_OPENGL_ES3_BIT,
+        EGL_RENDERABLE_TYPE,
+#ifdef _CE22_
+        EGL_OPENGL_ES3_BIT,
+#else
+        EGL_OPENGL_ES2_BIT,
+#endif
         EGL_NONE
     };
 
     EGLint count = 0;
 	eglGetConfigs(egl_dpy, NULL, 0, &count);
 	if (count < 1) {
 		printf("drmdevice: %s: no EGL configs to choose from\n", __FUNCTION__);
         return false;
