diff -ru8bBw a/video.c b/video.c
--- a/video.c	2024-03-25 17:07:13.000000000 +0100
+++ b/video.c	2024-04-18 18:37:23.503240219 +0200
@@ -1778,16 +1778,19 @@
 
 	amlSetInt("/sys/class/graphics/fb0/blank", 1);
 
 	if (myKernel == 5) {
 		char t[80];
 		sprintf(mode,"U:%dx%dp-0\n",VideoWindowWidth,VideoWindowHeight);
 		amlGetString("/sys/class/graphics/fb0/modes",t,sizeof(t)); // need to read the modes first
 		amlSetString("/sys/class/graphics/fb0/mode", mode);
+		amlSetInt("/sys/module/aml_media/parameters/di_debug_flag",0);
+		amlSetString("/sys/class/deinterlace/di0/debug","di_debug_flag0x0");
+		amlSetInt("/sys/module/amvdec_h264/parameters/dec_control",4);
 	}
 
 	fd = open("/dev/fb0", O_RDWR);
 
 	ioctl(fd, FBIOGET_VSCREENINFO, &info);
 	info.reserved[0] = 0;
 	info.reserved[1] = 0;
 	info.reserved[2] = 0;
