From 0bc824e1e2d238053acf0c753845bc386496226d Mon Sep 17 00:00:00 2001
From: Andreas Baierl <ichgeh@imkreisrum.de>
Date: Thu, 27 Oct 2022 17:07:58 +0200
Subject: [PATCH 1/2] add show_vdr_config option to show_config

---
 config/show_config | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff -bBwu8d a/config/show_config b/config/show_config
--- a/config/show_config	2022-11-03 15:39:24.814562546 +0100
+++ b/config/show_config	2022-11-03 15:41:33.589975126 +0100
@@ -139,15 +139,26 @@
 
   if [ -f distributions/${DISTRO}/show_config ]; then
     . distributions/${DISTRO}/show_config
   fi
   if [ "$(type -t show_distro_config)" = "function" ]; then
     show_distro_config
   fi
 
+  # Distribution specific VDR configuration
+  # show_vdr_distro_config() should be included in one of:
+  #   $DISTRO/show_config
+
+  if [ "${VDR}" != "no" ] && [ -f distributions/${DISTRO}/show_vdr_config ]; then
+    . distributions/${DISTRO}/show_vdr_config
+  fi
+  if [ "${VDR}" != "no" ] && [ "$(type -t show_vdr_distro_config)" = "function" ]; then
+    show_vdr_distro_config
+  fi
+
   config_message+="\n\n ${dashes}${dashes}${dashes}"
   config_message+="\n End Configuration for ${DISTRONAME}"
   config_message+="\n ${dashes}${dashes}${dashes}"
   config_message+="\n\n\n"
 
   echo -e "${config_message}"
 }
