diff --git a/projects/Amlogic-ce/devices/Amlogic-no/packages/linux/package.mk b/projects/Amlogic-ce/devices/Amlogic-no/packages/linux/package.mk
index a93ccc6c35..00277a4c6f 100644
--- a/projects/Amlogic-ce/devices/Amlogic-no/packages/linux/package.mk
+++ b/projects/Amlogic-ce/devices/Amlogic-no/packages/linux/package.mk
@@ -115,6 +115,9 @@ post_patch() {
       sed -e "s|^CONFIG_WIREGUARD=.*$|# CONFIG_WIREGUARD is not set|" -i ${PKG_BUILD}/.config
     fi
 
+    # enable meson vrtc driver
+    sed -e "s|^# CONFIG_RTC_DRV_MESON_VRTC is not set$|CONFIG_RTC_DRV_MESON_VRTC=m|" -i $PKG_BUILD/.config
+
     # 5.4.125 kernel compile errors
     sed -e 's|^KBUILD_CFLAGS += $(call cc-option,-Wimplicit-fallthrough,).*||' \
         -e 's|^KBUILD_CFLAGS   := \(.*\)|KBUILD_CFLAGS   := -Wno-format -Wno-unused-function -Wno-misleading-indentation \1|' \
