diff --git a/packages/multimedia/ffmpeg/package.mk b/packages/multimedia/ffmpeg/package.mk
index 6626100802..318d7fc047 100644
--- a/packages/multimedia/ffmpeg/package.mk
+++ b/packages/multimedia/ffmpeg/package.mk
@@ -235,6 +256,13 @@ configure_target() {
               ${PKG_FFMPEG_FPU} \
               --disable-symver \
               ${PKG_FFMPEG_TESTING}
+
+   # create version file
+	make libavutil/ffversion.h
+}
+
+make_target() {
+	make V=1 -j
 }
 
 post_makeinstall_target() {
