diff --git a/packages/multimedia/ffmpeg/package.mk b/packages/multimedia/ffmpeg/package.mk
index 6626100802..cbf735e328 100644
--- a/packages/multimedia/ffmpeg/package.mk
+++ b/packages/multimedia/ffmpeg/package.mk
@@ -237,6 +258,17 @@ configure_target() {
               ${PKG_FFMPEG_TESTING}
 }
 
+make_target() {
+	echo "==> Make ffversion.h"
+    make libavutil/ffversion.h
+    echo "==> cat vor dem Build"
+    cat libavutil/ffversion.h
+	make V=1 -j || true
+	echo "==> cat nach dem Build"
+	cat libavutil/ffversion.h
+	exit 1
+}
+
 post_makeinstall_target() {
   rm -rf ${INSTALL}/usr/share/ffmpeg/examples
 }
