diff --git a/packages/multimedia/ffmpeg/package.mk b/packages/multimedia/ffmpeg/package.mk
index 6626100802..5059d30fcc 100644
--- a/packages/multimedia/ffmpeg/package.mk
+++ b/packages/multimedia/ffmpeg/package.mk
@@ -237,6 +248,10 @@ configure_target() {
               ${PKG_FFMPEG_TESTING}
 }
 
+make_target() {
+	make V=1 -j
+}
+
 post_makeinstall_target() {
   rm -rf ${INSTALL}/usr/share/ffmpeg/examples
 }
