diff --git a/packages/python/devel/pluggy/package.mk b/packages/python/devel/pluggy/package.mk
index 9ff5435fe3..87bb7b5ba1 100644
--- a/packages/python/devel/pluggy/package.mk
+++ b/packages/python/devel/pluggy/package.mk
@@ -2,8 +2,8 @@
 # Copyright (C) 2025-present Team LibreELEC (https://libreelec.tv)
 
 PKG_NAME="pluggy"
-PKG_VERSION="fc40456932ec7fb258b0c48ef3789b7cd7747647"
-PKG_SHA256="614a8426baa09a837cfa15bd5d842a4951e55c4f5b28d63fe87d3fe6677633a9"
+PKG_VERSION="1.5.0"
+PKG_SHA256="6e9b6c46ae3b0c82087df05d866a8ba2787480dfb2c93b51b78ffc15deaf51b9"
 PKG_LICENSE="MIT"
 PKG_SITE="https://github.com/pytest-dev/pluggy"
 PKG_URL="https://github.com/pytest-dev/pluggy/archive/${PKG_VERSION}.tar.gz"
