diff --git a/packages/python/devel/meson/package.mk b/packages/python/devel/meson/package.mk
index 7e9adf1803..7445014235 100644
--- a/packages/python/devel/meson/package.mk
+++ b/packages/python/devel/meson/package.mk
@@ -2,8 +2,8 @@
 # Copyright (C) 2016-present Team LibreELEC (https://libreelec.tv)
 
 PKG_NAME="meson"
-PKG_VERSION="1.10.0"
-PKG_SHA256="8071860c1f46a75ea34801490fd1c445c9d75147a65508cd3a10366a7006cc1c"
+PKG_VERSION="1.9.2"
+PKG_SHA256="3499b59bb23982496e01e57b4103ac2f826f9c3a3f59e507a0a832487fe55e3d"
 PKG_LICENSE="Apache"
 PKG_SITE="https://mesonbuild.com"
 PKG_URL="https://github.com/mesonbuild/meson/releases/download/${PKG_VERSION}/${PKG_NAME}-${PKG_VERSION}.tar.gz"
