diff --git a/openglosd.cpp b/openglosd.cpp
index 8e51368..4bd65ae 100644
--- a/openglosd.cpp
+++ b/openglosd.cpp
@@ -1271,8 +1271,8 @@ bool cOglCmdDrawImage::Execute(void) {
 
     GLfloat x1 = x;          //left
     GLfloat y1 = y;          //top
-    GLfloat x2 = x + width;  //right
-    GLfloat y2 = y + height; //bottom
+    GLfloat x2 = x + width * scaleX;    //right
+    GLfloat y2 = y + height * scaleY;   //bottom
 
     GLfloat quadVertices[] = {
         x1, y2,   0.0, 1.0,     // left bottom
@@ -1775,7 +1775,7 @@ void cOglPixmap::DrawImage(const cPoint &Point, const cImage &Image) {
         return;
     memcpy(argb, Image.Data(), sizeof(tColor) * Image.Width() * Image.Height());
 
-    oglThread->DoCmd(new cOglCmdDrawImage(fb, argb, Image.Width(), Image.Height(), Point.X(), Point.Y()));
+    oglThread->DoCmd(new cOglCmdDrawImage(fb, argb, Image.Width(), Image.Height(), Point.X(), Point.Y(), true, Image.FactorX(), Image.FactorY()));
 
     SetDirty();
     MarkDrawPortDirty(cRect(Point, cSize(Image.Width(), Image.Height())).Intersected(DrawPort().Size()));
