diff --git a/menusched.c b/menusched.c
index 3e4bbee..3cbd2a8 100644
--- a/menusched.c
+++ b/menusched.c
@@ -781,7 +781,7 @@ int cMenuEpgWhatsOn::LoadSearch(const cUserTimes::UserTime* userTime)
       const char* strChannelId = menuDb->useeventsDb->getStrValue("CHANNELID");
       const cChannel* channel = channels->GetByChannelID(tChannelID::FromString(strChannelId));
       const cSchedule* schedule = schedules->GetSchedule(channel);
-      const cEvent* event = !schedule ? 0 :  schedule->GetEvent(menuDb->useeventsDb->getIntValue("USEID"));
+      const cEvent* event = !schedule ? 0 :  schedule->GetEventById(menuDb->useeventsDb->getIntValue("USEID"));
 
       if (event)
          Add(new cMenuEpgScheduleItem(menuDb, event, channel, yes));
diff --git a/recording.c b/recording.c
index 44f012e..adaa813 100644
--- a/recording.c
+++ b/recording.c
@@ -432,7 +432,7 @@ int cUpdate::updateRecordingTable(int fullReload)
                cStateKey schedulesKey;
                const cSchedules* schedules = cSchedules::GetSchedulesRead(schedulesKey, 500/*ms*/);
                const cSchedule* s = schedules ? (cSchedule*)schedules->GetSchedule(channel) : 0;
-               const cEvent* event = s ? s->GetEvent(eventId) : 0;
+               const cEvent* event = s ? s->GetEventById(eventId) : 0;
 
                if (event && !isEmpty(event->Aux()) && xml.set(event->Aux()) == success)
                {
diff --git a/timer.c b/timer.c
index 6e6f63f..fc5b894 100644
--- a/timer.c
+++ b/timer.c
@@ -301,7 +301,7 @@ int cUpdate::performTimerJobs()
             continue;
          }
 
-         if (eventid > 0 && !(event = s->GetEvent(eventid)))
+         if (eventid > 0 && !(event = s->GetEventById(eventid)))
          {
             const cChannel* channel = channels->GetByChannelID(channelId);
 
diff --git a/ttools.c b/ttools.c
index 21530fd..1d6ef76 100644
--- a/ttools.c
+++ b/ttools.c
@@ -466,7 +466,7 @@ cEpgTimer* newTimerObjectFromRow(cDbRow* timerRow, cDbRow* vdrRow)
          const cSchedule* schedule = schedules->GetSchedule(channel);
 
          if (schedule)
-            event = schedule->GetEvent(timerRow->getIntValue("EVENTID"));
+            event = schedule->GetEventById(timerRow->getIntValue("EVENTID"));
       }
 
 #if defined (APIVERSNUM) && (APIVERSNUM >= 20301)
diff --git a/update.c b/update.c
index 3f791dc..02dc60c 100644
--- a/update.c
+++ b/update.c
@@ -1646,7 +1646,7 @@ int cUpdate::refreshEpg(const char* forChannelId, int maxTries)
 
             // get event / timer
 
-            if ((event = s->GetEvent(eventsDb->getIntValue("USEID"))))
+            if ((event = s->GetEventById(eventsDb->getIntValue("USEID"))))
             {
                if (Us::isRemove(updFlg))
                   tell(2, "Remove event %uld of channel '%s' due to updflg %c",
