diff --git a/packages/multimedia/ffmpeg/package.mk b/packages/multimedia/ffmpeg/package.mk
index b55ffbd2c8..300e03e480 100644
--- a/packages/multimedia/ffmpeg/package.mk
+++ b/packages/multimedia/ffmpeg/package.mk
@@ -8,7 +8,7 @@ PKG_SHA256="9b16b8731d78e596b4be0d720428ca42df642bb2d78342881ff7f5bc29fc9623"
 PKG_LICENSE="GPL-3.0-only"
 PKG_SITE="https://ffmpeg.org"
 PKG_URL="http://ffmpeg.org/releases/ffmpeg-${PKG_VERSION}.tar.xz"
-PKG_DEPENDS_TARGET="toolchain zlib bzip2 openssl speex libxml2"
+PKG_DEPENDS_TARGET="toolchain zlib bzip2 openssl speex libxml2 libvpx libvorbis opus"
 PKG_LONGDESC="FFmpeg is a complete, cross-platform solution to record, convert and stream audio and video."
 PKG_PATCH_DIRS="libreelec"
 
@@ -127,7 +127,7 @@ if [ "${FFMPEG_TESTING}" = "yes" ]; then
     PKG_FFMPEG_TESTING+=" --enable-vout-drm --enable-outdev=vout_drm"
   fi
 else
-  PKG_FFMPEG_TESTING="--disable-programs"
+  PKG_FFMPEG_TESTING=""
 fi
 
 configure_target() {
@@ -190,6 +190,13 @@ configure_target() {
               --enable-encoder=aac \
               --enable-encoder=wmav2 \
               --enable-encoder=mjpeg \
+              --enable-libvpx \
+              --enable-encoder=libvpx_vp8 \
+              --enable-encoder=libvpx_vp9 \
+              --enable-libvorbis \
+              --enable-libopus \
+              --enable-encoder=libvorbis \
+              --enable-encoder=libopus --enable-encoder=mpeg2video --enable-encoder=mp2 \
               --enable-encoder=png \
               ${PKG_FFMPEG_HWACCEL} \
               --disable-muxers \
@@ -198,6 +205,11 @@ configure_target() {
               --enable-muxer=asf \
               --enable-muxer=ipod \
               --enable-muxer=mpegts \
+              --enable-muxer=matroska \
+              --enable-muxer=mp4 \
+              --enable-muxer=webm \
+              --enable-muxer=webm_chunk \
+              --enable-muxer=webm_dash_manifest \
               --enable-demuxers \
               --enable-parsers \
               --enable-bsfs \
@@ -223,8 +235,6 @@ configure_target() {
               --enable-libspeex \
               --disable-libtheora \
               --disable-libvo-amrwbenc \
-              --disable-libvorbis \
-              --disable-libvpx \
               --disable-libx264 \
               --disable-libxavs \
               --enable-libxml2 \
