diff --git a/packages/audio/opus/package.mk b/packages/audio/opus/package.mk
index d37e97f3d4..3f29e35bba 100644
--- a/packages/audio/opus/package.mk
+++ b/packages/audio/opus/package.mk
@@ -19,5 +19,5 @@ else
 fi
 
 PKG_CONFIGURE_OPTS_TARGET="--enable-static \
-                           --disable-shared \
+                           --enable-shared \
                            ${PKG_FIXED_POINT}"
