diff --git a/packages/graphics/kmscube/package.mk b/packages/graphics/kmscube/package.mk
index c858a2ce4f..e3a02edbfc 100644
--- a/packages/graphics/kmscube/package.mk
+++ b/packages/graphics/kmscube/package.mk
@@ -3,7 +3,7 @@
 
 PKG_NAME="kmscube"
 PKG_VERSION="9f63f359fab1b5d8e862508e4e51c9dfe339ccb0"
-PKG_SHA256="896ac76671a9b89836a7014b16cc85b45b041e03fe34a8f529f4718aa2b15cef"
+PKG_SHA256="5153f31e4e7da0938050c7d40fa6d5f2cc0a6d7a0c54d0da0ef76de535a36daa"
 PKG_LICENSE="GPL"
 PKG_SITE="https://gitlab.freedesktop.org/mesa/kmscube"
 PKG_URL="https://gitlab.freedesktop.org/mesa/kmscube/-/archive/master/kmscube-${PKG_VERSION}.tar.gz"
