diff -ru8Bbw a/extensions/libxmlwrapper.c b/extensions/libxmlwrapper.c
--- a/extensions/libxmlwrapper.c	2023-10-14 13:52:11.000000000 +0200
+++ b/extensions/libxmlwrapper.c	2023-12-29 14:50:25.395586095 +0100
@@ -1,22 +1,22 @@
 #include "helpers.h"
 #include "libxmlwrapper.h"
 
-void SkinDesignerXMLErrorHandler (void * userData, xmlErrorPtr error) {
+void SkinDesignerXMLErrorHandler (void * userData, const xmlErrorPtr error) {
     esyslog("skindesigner: Error in XML: %s", error->message);
 }
 
 cLibXMLWrapper::cLibXMLWrapper(void) {
     ctxt = NULL;
     doc = NULL;
     root = NULL;
 
-    initGenericErrorDefaultFunc(NULL);
-    xmlSetStructuredErrorFunc(NULL, SkinDesignerXMLErrorHandler);
+    xmlSetGenericErrorFunc(NULL, NULL);
+    xmlSetStructuredErrorFunc(NULL, (xmlStructuredErrorFunc)SkinDesignerXMLErrorHandler);
     ctxt = xmlNewParserCtxt();
 }
 
 cLibXMLWrapper::~cLibXMLWrapper() {
     DeleteDocument();
     xmlFreeParserCtxt(ctxt);
 }
 
