diff -rubBwd a/OpenVFDService.c b/OpenVFDService.c
--- a/OpenVFDService.c	2022-09-02 17:07:06.000000000 +0200
+++ b/OpenVFDService.c	2022-10-26 06:44:26.782438353 +0200
@@ -378,6 +378,19 @@
 						sync_data.useBuffer = true;
 					}
 					break;
+                case 3:
+                    if (ret >= 2) {
+                        sync_data.display_data.mode = DISPLAY_MODE_TITLE;
+                        sync_data.useBuffer = true;
+                        snprintf(sync_data.display_data.string_main, sizeof(sync_data.display_data.string_secondary), &buf[1]);
+                    }
+                    break;
+                case 4:
+                    if (ret >= 2) {
+                        sync_data.display_data.mode = DISPLAY_MODE_TITLE;
+                        sync_data.useBuffer = true;
+                        snprintf(sync_data.display_data.string_secondary, sizeof(sync_data.display_data.string_secondary), &buf[1]);
+                    }
 				}
 			}
 			if (!skipSignal)
