From 98df3325a80f0467a388ce67587f97074200e966 Mon Sep 17 00:00:00 2001
From: kamel5 <vdr.kamel5 (at) gmx (dot) net>
Date: Sat, 17 Dec 2022 13:07:57 +0100
Subject: [PATCH] Restore volume

---
 vdr.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/vdr.c b/vdr.c
index 3f88b180..dffcc161 100644
--- a/vdr.c
+++ b/vdr.c
@@ -929,10 +929,11 @@ int main(int argc, char *argv[])
     LOCK_CHANNELS_READ;
     Channels->SwitchTo(Setup.CurrentChannel);
   }
+
+  // Restore volume:
+  cDevice::PrimaryDevice()->SetVolume(Setup.CurrentVolume, true);
   if (MuteAudio)
      cDevice::PrimaryDevice()->ToggleMute();
-  else
-     cDevice::PrimaryDevice()->SetVolume(Setup.CurrentVolume, true);
 
   // Signal handlers:
 
-- 
2.38.1

