--- a/vdr.c       2026/01/21 21:19:59     5.23
+++ b/vdr.c       2026/01/29 19:40:45
@@ -1362,7 +1362,7 @@
           // Pausing live video:
           case kPlayPause:
           case kPause:
-               if (!Control) {
+               if (!cReplayControl::NowReplaying()) {
                   DELETE_MENU;
                   if (Setup.PauseKeyHandling) {
                      if (Setup.PauseKeyHandling > 1 || Interface->Confirm(tr("Pause live video?"))) {
@@ -1375,7 +1375,7 @@
                break;
           // Instant recording:
           case kRecord:
-               if (!Control) {
+               if (!cReplayControl::NowReplaying()) {
                   if (Setup.RecordKeyHandling) {
                      if (Setup.RecordKeyHandling > 1 || Interface->Confirm(tr("Start recording?"))) {
                         if (cRecordControls::Start())