--- vdr-2.6.7.orig/eitscan.c   2024/03/10 11:16:29     5.3
+++ vdr-2.6.7.orig/eitscan.c   2024/05/19 14:55:57
@@ -147,7 +147,9 @@
            for (int i = 0; i < cDevice::NumDevices(); i++) {
                cDevice *Device = cDevice::GetDevice(i);
                if (Device && Device->ProvidesEIT()) {
-                  for (cScanData *ScanData = scanList->First(); ScanData; ScanData = scanList->Next(ScanData)) {
+                  cScanData *Next = NULL;
+                  for (cScanData *ScanData = scanList->First(); ScanData; ScanData = Next) {
+                      Next = scanList->Next(ScanData);
                       const cChannel *Channel = ScanData->GetChannel();
                       if (Channel) {
                          if (Device->IsTunedToTransponder(Channel))