--- a/tools.c     2024/06/21 06:27:20     5.11
+++ b/tools.c     2024/07/10 14:50:07
@@ -169,8 +169,10 @@
               }
            }
         char *sof = s + of;
-        if (l2 != l1)
+        if (l2 != l1) {
            memmove(sof + l2, sof + l1, l - of - l1 + 1);
+           l += l2 - l1;
+           }
         memcpy(sof, s2, l2);
         q = sof + l2;
         } while (p = strstr(q, s1));