--- ./recording.c	2022/11/06 11:26:16	5.15
+++ ./recording.c	2022/11/13 14:49:08
@@ -2413,6 +2413,7 @@
         isyslog("generating index file");
      }
   Skins.QueueMessage(mtInfo, tr("Regenerating index file"));
+  SetRecordingTimerId(recordingName, cString::sprintf("%d@%s", 0, Setup.SVDRPHostName));
   bool Stuffed = false;
   while (Running()) {
         // Rewind input file:
@@ -2506,6 +2507,7 @@
            break;
            }
         }
+  SetRecordingTimerId(recordingName, NULL);
   if (IndexFileComplete) {
      if (IndexFileWritten) {
         cRecordingInfo RecordingInfo(recordingName);
