diff --git a/config.c b/config.c
index 23d9e1ee..16aa41c3 100644
--- a/config.c
+++ b/config.c
@@ -435,6 +435,8 @@ cSetup::cSetup(void)
   AlwaysSortFoldersFirst = 1;
   DefaultSortModeRec = rsmTime;
   RecSortingDirection = rsdAscending;
+  UndeleteTimeOut = 0;
+  ButtonFirst = 0;
   NumberKeysForChars = 1;
   ColorKey0 = 0;
   ColorKey1 = 1;
@@ -665,6 +667,8 @@ bool cSetup::Parse(const char *Name, const char *Value)
   else if (!strcasecmp(Name, "AlwaysSortFoldersFirst")) AlwaysSortFoldersFirst = atoi(Value);
   else if (!strcasecmp(Name, "RecSortingDirection")) RecSortingDirection= atoi(Value);
   else if (!strcasecmp(Name, "DefaultSortModeRec"))  DefaultSortModeRec = atoi(Value);
+  else if (!strcasecmp(Name, "UndeleteTimeOut"))     UndeleteTimeOut    = atoi(Value);
+  else if (!strcasecmp(Name, "ButtonFirst"))         ButtonFirst        = atoi(Value);
   else if (!strcasecmp(Name, "NumberKeysForChars"))  NumberKeysForChars = atoi(Value);
   else if (!strcasecmp(Name, "ColorKey0"))           ColorKey0          = atoi(Value);
   else if (!strcasecmp(Name, "ColorKey1"))           ColorKey1          = atoi(Value);
@@ -799,6 +803,8 @@ bool cSetup::Save(void)
   Store("AlwaysSortFoldersFirst", AlwaysSortFoldersFirst);
   Store("RecSortingDirection",RecSortingDirection);
   Store("DefaultSortModeRec", DefaultSortModeRec);
+  Store("UndeleteTimeOut",    UndeleteTimeOut);
+  Store("ButtonFirst",        ButtonFirst);
   Store("NumberKeysForChars", NumberKeysForChars);
   Store("ColorKey0",          ColorKey0);
   Store("ColorKey1",          ColorKey1);
diff --git a/config.h b/config.h
index 3eeb7084..23365d7b 100644
--- a/config.h
+++ b/config.h
@@ -316,6 +316,8 @@ public:
   int AlwaysSortFoldersFirst;
   int DefaultSortModeRec;
   int RecSortingDirection;
+  int UndeleteTimeOut;
+  int ButtonFirst;
   int NumberKeysForChars;
   int ColorKey0, ColorKey1, ColorKey2, ColorKey3;
   int VideoDisplayFormat;
diff --git a/menu.c b/menu.c
index 8c1c478a..f3e3f16f 100644
--- a/menu.c
+++ b/menu.c
@@ -2902,14 +2902,15 @@ private:
   cString originalFileName;
   cStateKey recordingsStateKey;
   bool withButtons;
+  bool deletedRecording;
   bool RefreshRecording(void);
 public:
-  cMenuRecording(const cRecording *Recording, bool WithButtons = false);
+  cMenuRecording(const cRecording *Recording, bool WithButtons = false, bool DeletedRecording = false);
   virtual void Display(void);
   virtual eOSState ProcessKey(eKeys Key);
 };
 
-cMenuRecording::cMenuRecording(const cRecording *Recording, bool WithButtons)
+cMenuRecording::cMenuRecording(const cRecording *Recording, bool WithButtons, bool DeletedRecording)
 :cOsdMenu(tr("Recording info"))
 {
   SetMenuCategory(mcRecordingInfo);
@@ -2918,13 +2919,17 @@ cMenuRecording::cMenuRecording(const cRecording *Recording, bool WithButtons)
   recording = Recording;
   originalFileName = recording->FileName();
   withButtons = WithButtons;
+  deletedRecording = DeletedRecording;
   if (withButtons)
+     if (deletedRecording)
+        SetHelp(tr("Button$RECORDINGS"), NULL, NULL, tr("Button$Back"));
+     else
      SetHelp(tr("Button$Play"), tr("Button$Rewind"), NULL, tr("Button$Edit"));
 }
 
 bool cMenuRecording::RefreshRecording(void)
 {
-  if (const cRecordings *Recordings = cRecordings::GetRecordingsRead(recordingsStateKey)) {
+  if (const cRecordings *Recordings = deletedRecording ? cRecordings::GetDeletedRecordingsRead(recordingsStateKey) : cRecordings::GetRecordingsRead(recordingsStateKey)) {
      if ((recording = Recordings->GetByName(originalFileName)) != NULL)
         Display();
      else {
@@ -2975,15 +2980,28 @@ eOSState cMenuRecording::ProcessKey(eKeys Key)
 
   if (state == osUnknown) {
      switch (Key) {
-       case kRed:    if (withButtons)
-                        Key = kOk; // will play the recording, even if recording commands are defined
-       case kGreen:  if (!withButtons)
+       case kRed:    if (withButtons) {
+                        if (!deletedRecording)
+                           Key = kOk;
+                        cRemote::Put(Key, true); // will play the recording, even if recording commands are defined
+                        return osBack;
+                        }
+                     break;
+       case kGreen:  if (withButtons) {
+                        if (deletedRecording)
+                           state = osContinue;
+                        else {
+                           state = osBack;
+                           cRemote::Put(Key, true); // continue with osBack to close the info menu and process the key
+                           }
+                        return state;
+                        }
                         break;
-                     cRemote::Put(Key, true);
-                     // continue with osBack to close the info menu and process the key
+       case kYellow: return osContinue;
        case kOk:     return osBack;
        case kBlue:   if (withButtons)
-                        return AddSubMenu(new cMenuRecordingEdit(recording));
+                        return deletedRecording ? osBack
+                                                : AddSubMenu(new cMenuRecordingEdit(recording));
                      break;
        default: break;
        }
@@ -3051,14 +3069,17 @@ void cMenuRecordingItem::SetMenuItem(cSkinDisplayMenu *DisplayMenu, int Index, b
 cString cMenuRecordings::path;
 cString cMenuRecordings::fileName;
 
-cMenuRecordings::cMenuRecordings(const char *Base, int Level, bool OpenSubMenus, const cRecordingFilter *Filter)
-:cOsdMenu(Base ? Base : tr("Recordings"), 9, 6, 6)
+cMenuRecordings::cMenuRecordings(const char *Base, int Level, bool OpenSubMenus, const cRecordingFilter *Filter, bool DelRecMenu)
+:cOsdMenu(Base ? Base : DelRecMenu ? tr("Deleted Recordings") : tr("Recordings"), 9, 6, 6)
 {
   SetMenuCategory(mcRecording);
   base = Base ? strdup(Base) : NULL;
   level = Setup.RecordingDirs ? Level : -1;
   filter = Filter;
   helpKeys = -1;
+  timeOut = 0;
+  delRecMenu = DelRecMenu;
+  deletedRec = false;
   Display(); // this keeps the higher level menus from showing up briefly when pressing 'Back' during replay
   Set();
   if (Current() < 0)
@@ -3079,38 +3100,55 @@ cMenuRecordings::~cMenuRecordings()
         SetRecording(ri->Recording()->FileName());
      }
   free(base);
+  LOCK_RECORDINGS_WRITE;
 }
 
 void cMenuRecordings::SetHelpKeys(void)
 {
   cMenuRecordingItem *ri = (cMenuRecordingItem *)Get(Current());
   int NewHelpKeys = 0;
+  bool DeletedRec = false;
+  bool commands = false;
   if (ri) {
      if (ri->IsDirectory())
         NewHelpKeys = 1;
-     else
+     else {
+        commands = RecordingCommands.Count() ? true : false;
         NewHelpKeys = 2;
      }
-  if (NewHelpKeys != helpKeys) {
+     }
+  {
+  LOCK_DELETEDRECORDINGS_READ;
+  if (!commands || !Setup.UndeleteTimeOut || (!Setup.ButtonFirst && timeOut < Setup.UndeleteTimeOut) || (Setup.ButtonFirst && timeOut >= Setup.UndeleteTimeOut))
+     DeletedRec = DeletedRecordings->Count() ? true : false;
+  }
+  if (delRecMenu)
+     switch (NewHelpKeys) {
+       case 0: SetHelp(NULL); break;
+       case 1: SetHelp(tr("Button$RECORDINGS")); break;
+       case 2: SetHelp(tr("Button$RECORDINGS"), tr("Button$Undelete"), tr("Button$Destroy"),  tr("Button$Info"));
+     }
+  else if (NewHelpKeys != helpKeys || DeletedRec != deletedRec) {
      switch (NewHelpKeys) {
        case 0: SetHelp(NULL); break;
-       case 1: SetHelp(tr("Button$Open"), NULL, NULL, tr("Button$Edit")); break;
-       case 2: SetHelp(RecordingCommands.Count() ? tr("Commands") : tr("Button$Play"), tr("Button$Rewind"), tr("Button$Delete"), tr("Button$Info"));
+       case 1: SetHelp(DeletedRec ? tr("Button$UNDELETE") : tr("Button$Open"), NULL, NULL, tr("Button$Edit")); break;
+       case 2: SetHelp(DeletedRec ? tr("Button$UNDELETE") : RecordingCommands.Count() ? tr("Commands") : tr("Button$Play"), tr("Button$Rewind"), tr("Button$Delete"), tr("Button$Info"));
        default: ;
        }
      helpKeys = NewHelpKeys;
+     deletedRec = DeletedRec;
      }
 }
 
 void cMenuRecordings::Set(bool Refresh)
 {
-  if (cRecordings::GetRecordingsRead(recordingsStateKey)) {
+  if (delRecMenu ? cRecordings::GetDeletedRecordingsRead(recordingsStateKey) : cRecordings::GetRecordingsRead(recordingsStateKey)) {
      recordingsStateKey.Remove();
-     cRecordings *Recordings = cRecordings::GetRecordingsWrite(recordingsStateKey); // write access is necessary for sorting!
+     cRecordings *Recordings = delRecMenu ? cRecordings::GetDeletedRecordingsWrite(recordingsStateKey) : cRecordings::GetRecordingsWrite(recordingsStateKey); // write access is necessary for sorting!
      const char *CurrentRecording = NULL;
      if (cMenuRecordingItem *ri = (cMenuRecordingItem *)Get(Current()))
         CurrentRecording = ri->Recording()->FileName();
-     if (!CurrentRecording)
+     if (!CurrentRecording && !delRecMenu)
         CurrentRecording = *fileName ? *fileName : cReplayControl::LastReplayed();
      int current = Current();
      Clear();
@@ -3187,7 +3225,7 @@ bool cMenuRecordings::Open(bool OpenSubMenus)
         buffer = cString::sprintf("%s%c%s", base, FOLDERDELIMCHAR, t);
         t = buffer;
         }
-     AddSubMenu(new cMenuRecordings(t, level + 1, OpenSubMenus, filter));
+     AddSubMenu(new cMenuRecordings(t, level + 1, OpenSubMenus, filter, delRecMenu));
      return true;
      }
   return false;
@@ -3320,6 +3358,83 @@ eOSState cMenuRecordings::Delete(void)
   return osContinue;
 }
 
+eOSState cMenuRecordings::Undelete(void)
+{
+#define RECEXT       ".rec"
+#define DELEXT       ".del"
+  if (HasSubMenu() || Count() == 0)
+     return osContinue;
+  if (cMenuRecordingItem *ri = (cMenuRecordingItem *)Get(Current())) {
+     if (ri->IsDirectory())
+        return osContinue;
+     else {
+        if (Interface->Confirm(tr("Restore recording?"))) {
+           cRecording *Recording = ((cRecording *)ri->Recording());
+           if (Recording) {
+              if (Recording->Undelete()) {
+                 cOsdMenu::Del(Current());
+                 if (cRecordings *DeletedRecordings = cRecordings::GetDeletedRecordingsWrite(recordingsStateKey)) { // write access is necessary for sorting!
+                    DeletedRecordings->Del(Recording);
+                    recordingsStateKey.Remove(true);
+                    char *NewName = strdup(Recording->FileName());
+                    char *ext = strrchr(NewName, '.');
+                    if (ext && strcmp(ext, DELEXT) == 0) {
+                       strncpy(ext, RECEXT, strlen(ext));
+                       LOCK_RECORDINGS_WRITE;
+                       Recordings->AddByName(NewName);
+                       cVideoDiskUsage::ForceCheck();
+                       }
+                    }
+                 else
+                    Skins.Message(mtError, tr("Error while restoring recording!"));
+                 Display();
+                 if (!Count())
+                    return osUserRecEmpty;
+                 return osUserRecRemoved;
+                 }
+              else
+                 Skins.Message(mtError, tr("Error while restoring recording!"));
+              }
+           }
+        }
+     }
+  return osContinue;
+}
+
+eOSState cMenuRecordings::Erase(void)
+{
+  dsyslog ("%s %s %d\n", __FILE__, __func__,  __LINE__);
+  if (HasSubMenu() || Count() == 0)
+     return osContinue;
+  if (cMenuRecordingItem *ri = (cMenuRecordingItem *)Get(Current())) {
+     if (ri->IsDirectory())
+        return osContinue;
+     else {
+        if (Interface->Confirm(tr("Delete recording?"))) {
+           cRecording *recording = ((cRecording *)ri->Recording());
+           if (recording) {
+              if (recording->Remove()) {
+                 cOsdMenu::Del(Current());
+                 if (cRecordings *DeletedRecordings = cRecordings::GetDeletedRecordingsWrite(recordingsStateKey)) { // write access is necessary for sorting!
+                    DeletedRecordings->Del(recording);
+                    recordingsStateKey.Remove(true);
+                    }
+                 else
+                    Skins.Message(mtError, tr("Error while removing recording!"));
+                 Display();
+                 if (!Count())
+                    return osUserRecEmpty;
+                 return osUserRecRemoved;
+                 }
+              else
+                 Skins.Message(mtError, tr("Error while removing recording!"));
+              }
+           }
+        }
+     }
+  return osContinue;
+}
+
 eOSState cMenuRecordings::Info(void)
 {
   if (HasSubMenu() || Count() == 0)
@@ -3328,7 +3443,7 @@ eOSState cMenuRecordings::Info(void)
      if (ri->IsDirectory())
         return AddSubMenu(new cMenuPathEdit(cString(ri->Recording()->Name(), strchrn(ri->Recording()->Name(), FOLDERDELIMCHAR, ri->Level() + 1))));
      else
-        return AddSubMenu(new cMenuRecording(ri->Recording(), true));
+        return AddSubMenu(new cMenuRecording(ri->Recording(), true, delRecMenu));
      }
   return osContinue;
 }
@@ -3368,14 +3483,15 @@ eOSState cMenuRecordings::ProcessKey(eKeys Key)
      switch (Key) {
        case kPlayPause:
        case kPlay:
-       case kOk:     return Play();
-       case kRed:    return (helpKeys > 1 && RecordingCommands.Count()) ? Commands() : Play();
-       case kGreen:  return Rewind();
-       case kYellow: return Delete();
+       case kOk:     if (delRecMenu) {Open(); return osContinue;} else return Play();
+       case kRed:    if (delRecMenu) {CloseSubMenu(false); delRecMenu = false; if (base) return state; else return osBack;}
+                     else return deletedRec ? AddSubMenu(new cMenuRecordings(NULL, 0, false, NULL, true)) : (helpKeys > 1 && RecordingCommands.Count()) ? Commands() : Play();
+       case kGreen:  return delRecMenu ? Undelete() : Rewind();
+       case kYellow: return delRecMenu ? Erase() : Delete();
        case kInfo:
        case kBlue:   return Info();
        case k0:      return Sort();
-       case k1...k9: return Commands(Key);
+       case k1...k9: return delRecMenu ? osContinue : Commands(Key);
        default: break;
        }
      }
@@ -3413,16 +3529,29 @@ eOSState cMenuRecordings::ProcessKey(eKeys Key)
      CloseSubMenu(false); // this is the now empty submenu
      cOsdMenu::Del(Current()); // the menu entry of the now empty subfolder
      Set(); // in case a recording was moved into a new subfolder of this folder
-     if (base && !Count()) // base: don't go up beyond the top level Recordings menu
+     if ((base || delRecMenu) && !Count()) // base: don't go up beyond the top level Recordings menu
         return state;
      Display();
      state = osContinue;
      }
   if (!HasSubMenu()) {
      Set(true);
-     if (Key != kNone)
+     if (Key != kNone) {
+        timeOut = 0;
         SetHelpKeys();
      }
+     if (!delRecMenu && Setup.UndeleteTimeOut) {
+        int delRecordings;
+        {
+        LOCK_DELETEDRECORDINGS_READ;
+        delRecordings = DeletedRecordings->Count();
+        }
+        if (delRecordings && timeOut < Setup.UndeleteTimeOut) {
+           timeOut++;
+           SetHelpKeys();
+           }
+        }
+     }
   return state;
 }
 
@@ -3456,6 +3585,7 @@ private:
   const char *recSortModeTexts[2];
   const char *recSortDirTexts[2];
   const char *keyColorTexts[4];
+  const char *buttonFirst[2];
   int osdLanguageIndex;
   int numSkins;
   int originalSkinIndex;
@@ -3511,6 +3641,8 @@ void cMenuSetupOSD::Set(void)
   recSortModeTexts[1] = tr("by time");
   recSortDirTexts[0] = tr("ascending");
   recSortDirTexts[1] = tr("descending");
+  buttonFirst[0] = tr("Button$UNDELETE");
+  buttonFirst[1] = tr("Key$Commands");
   keyColorTexts[0] = tr("Key$Red");
   keyColorTexts[1] = tr("Key$Green");
   keyColorTexts[2] = tr("Key$Yellow");
@@ -3546,6 +3678,9 @@ void cMenuSetupOSD::Set(void)
   Add(new cMenuEditBoolItem(tr("Setup.OSD$Always sort folders first"), &data.AlwaysSortFoldersFirst));
   Add(new cMenuEditStraItem(tr("Setup.OSD$Default sort mode for recordings"), &data.DefaultSortModeRec, 2, recSortModeTexts));
   Add(new cMenuEditStraItem(tr("Setup.OSD$Sorting direction for recordings"), &data.RecSortingDirection, 2, recSortDirTexts));
+  Add(new cMenuEditIntItem( tr("Setup.OSD$Red button UNDELETE timeout (s)"), &data.UndeleteTimeOut, 0, 10, tr("off")));
+  if (data.UndeleteTimeOut)
+  Add(new cMenuEditStraItem(tr("Setup.OSD$Displayed first (UNDELETE or Commands)"), &data.ButtonFirst, 2, buttonFirst));
   Add(new cMenuEditBoolItem(tr("Setup.OSD$Number keys for characters"), &data.NumberKeysForChars));
   Add(new cMenuEditStraItem(tr("Setup.OSD$Color key 0"),            &data.ColorKey0, 4, keyColorTexts));
   Add(new cMenuEditStraItem(tr("Setup.OSD$Color key 1"),            &data.ColorKey1, 4, keyColorTexts));
@@ -3595,12 +3730,13 @@ eOSState cMenuSetupOSD::ProcessKey(eKeys Key)
 
   int oldSkinIndex = skinIndex;
   int oldOsdLanguageIndex = osdLanguageIndex;
+  int oldUndeleteTimeOut = data.UndeleteTimeOut;
   eOSState state = cMenuSetupBase::ProcessKey(Key);
 
   if (ModifiedAppearance)
      cOsdProvider::UpdateOsdSize(true);
 
-  if (osdLanguageIndex != oldOsdLanguageIndex || skinIndex != oldSkinIndex) {
+  if (osdLanguageIndex != oldOsdLanguageIndex || skinIndex != oldSkinIndex || data.UndeleteTimeOut != oldUndeleteTimeOut) {
      strn0cpy(data.OSDLanguage, I18nLocale(osdLanguageIndex), sizeof(data.OSDLanguage));
      int OriginalOSDLanguage = I18nCurrentLanguage();
      I18nSetLanguage(osdLanguageIndex);
diff --git a/menu.h b/menu.h
index 88583028..4363fbc4 100644
--- a/menu.h
+++ b/menu.h
@@ -213,6 +213,9 @@ private:
   int level;
   cStateKey recordingsStateKey;
   int helpKeys;
+  int timeOut;
+  bool delRecMenu;
+  bool deletedRec;
   const cRecordingFilter *filter;
   static cString path;
   static cString fileName;
@@ -222,13 +225,15 @@ private:
   eOSState Play(void);
   eOSState Rewind(void);
   eOSState Delete(void);
+  eOSState Undelete(void);
+  eOSState Erase(void);
   eOSState Info(void);
   eOSState Sort(void);
   eOSState Commands(eKeys Key = kNone);
 protected:
   cString DirectoryName(void);
 public:
-  cMenuRecordings(const char *Base = NULL, int Level = 0, bool OpenSubMenus = false, const cRecordingFilter *Filter = NULL);
+  cMenuRecordings(const char *Base = NULL, int Level = 0, bool OpenSubMenus = false, const cRecordingFilter *Filter = NULL, bool DelRecMenu = false);
   ~cMenuRecordings();
   virtual eOSState ProcessKey(eKeys Key);
   static void SetRecording(const char *FileName);
diff --git a/po/de_DE.po b/po/de_DE.po
index 9880f311..421ce51e 100644
--- a/po/de_DE.po
+++ b/po/de_DE.po
@@ -867,15 +867,33 @@ msgstr "Fehler beim 
 msgid "Recording info"
 msgstr "Aufzeichnung"
 
+msgid "Button$RECORDINGS"
+msgstr "AUFNAHMEN"
+
+msgid "Button$Back"
+msgstr "Zurück"
+
 msgid "Button$Play"
 msgstr "Wiedergabe"
 
 msgid "Button$Rewind"
 msgstr "Anfang"
 
+msgid "Deleted Recordings"
+msgstr "gelöschte Aufnahmen"
+
 msgid "Recordings"
 msgstr "Aufzeichnungen"
 
+msgid "Button$Undelete"
+msgstr "Wiederherstellen"
+
+msgid "Button$Destroy"
+msgstr "Löschen"
+
+msgid "Button$UNDELETE"
+msgstr ""
+
 msgid "Commands"
 msgstr "Befehle"
 
@@ -888,6 +906,15 @@ msgstr "Aufzeichnung wird bearbeitet - trotzdem l
 msgid "Error while deleting recording!"
 msgstr "Fehler beim Löschen der Aufzeichnung!"
 
+msgid "Restore recording?"
+msgstr "Aufnahme wiederherstellen?"
+
+msgid "Error while restoring recording!"
+msgstr "Fehler bei der Wiederherstellung der Aufnahme!"
+
+msgid "Error while removing recording!"
+msgstr "Fehler beim Löschen der Aufnahme!"
+
 msgid "Recording commands"
 msgstr "Befehle für Aufzeichnungen"
 
@@ -1005,6 +1032,12 @@ msgstr "Standard Sortierreihenfolge f
 msgid "Setup.OSD$Sorting direction for recordings"
 msgstr "Sortierreihenfolge für Aufnahmen"
 
+msgid "Setup.OSD$Red button UNDELETE timeout (s)"
+msgstr ""
+
+msgid "Setup.OSD$Displayed first (UNDELETE or Commands)"
+msgstr "Wird zuerst angezeigt (UNDELETE oder Befehle)"
+
 msgid "Setup.OSD$Number keys for characters"
 msgstr "Nummerntasten für Zeichen"
 
