--- a/vdr.c       2022/11/28 10:44:01     5.10
+++ b/vdr.c       2022/12/12 10:18:06
@@ -651,13 +651,6 @@
   if (SysLogLevel > 0)
      openlog("vdr", LOG_CONS, SysLogTarget); // LOG_PID doesn't work as expected under NPTL

-  // Check the video directory:
-
-  if (!DirectoryOk(VideoDirectory, true)) {
-     fprintf(stderr, "vdr: can't access video directory %s\n", VideoDirectory);
-     return 2;
-     }
-
   // Daemon mode:

   if (DaemonMode) {
@@ -692,6 +685,13 @@
         }
      }

+  // Check the video directory:
+
+  if (!DirectoryOk(VideoDirectory, true)) {
+     fprintf(stderr, "vdr: can't access video directory %s\n", VideoDirectory);
+     return 2;
+     }
+
   isyslog("VDR version %s started", VDRVERSION);
   if (StartedAsRoot && VdrUser)
      isyslog("switched to user '%s'", VdrUser);