--- a/recording.c 2022/11/28 14:39:23
+++ b/recording.c 2022/12/01 12:39:42
@@ -2579,7 +2579,7 @@
               cCondWait::SleepMs(INDEXFILETESTINTERVAL);
         }
      int delta = 0;
-     if (!Record && (access(fileName, R_OK) != 0 || FileSize(fileName) == 0)) {
+     if (!Record && (access(fileName, R_OK) != 0 || FileSize(fileName) == 0 && time(NULL) - LastModifiedTime(fileName) > MAXWAITFORINDEXFILE)) {
         // Index file doesn't exist, so try to regenerate it:
         if (!isPesRecording) { // sorry, can only do this for TS recordings
            resumeFile.Delete(); // just in case