--- a/menu.c      2022/11/22 15:53:07
+++ b/menu.c      2022/12/01 12:55:34
@@ -3250,7 +3250,7 @@
         int Id;
         char *RemoteBuf = NULL;
         cString Remote;
-        if (2 == sscanf(TimerId, "%d@%m[^ \n]", &Id, &RemoteBuf)) {
+        if (2 == sscanf(TimerId, "%d@%m[^ \n]", &Id, &RemoteBuf) && Id != 0) {
            Remote = RemoteBuf;
            free(RemoteBuf);
            if (Interface->Confirm(tr("Timer still recording - really delete?"))) {