diff -u8bBwd a/Makefile b/Makefile
--- a/Makefile	2021-02-17 23:42:19.000000000 +0100
+++ b/Makefile	2022-10-15 17:57:05.223263152 +0200
@@ -1,9 +1,9 @@
-PREFIX:=/usr/
+PREFIX:=/usr/local
 DESTDIR:=/
 BINDIR:=$(DESTDIR)/$(PREFIX)/sbin/
 MANDIR:=$(DESTDIR)/$(PREFIX)/share/man/man1/
 
 PKGCONFIG = pkg-config
 HAVE_PKGCONFIG = $(shell $(PKGCONFIG) --version 2>/dev/null || echo no)
 ifneq ($(HAVE_PKGCONFIG),no)
 HAVE_SYSTEMD = $(shell $(PKGCONFIG) --exists libsystemd && echo 1 || echo 0)
