From 4080828fb7cd1fee5ffa75a17fbfe7b8278fbd2f Mon Sep 17 00:00:00 2001
From: Andreas Baierl <ichgeh@imkreisrum.de>
Date: Wed, 26 Jul 2023 11:05:54 +0200
Subject: [PATCH] build ffmpeg programs

---

diff --git a/packages/audio/libvorbis/package.mk b/packages/audio/libvorbis/package.mk
index 2b72481931..72d6a15f67 100644
--- a/packages/audio/libvorbis/package.mk
+++ b/packages/audio/libvorbis/package.mk
@@ -14,7 +14,7 @@ PKG_TOOLCHAIN="autotools"
 PKG_BUILD_FLAGS="+pic"
 
 PKG_CONFIGURE_OPTS_TARGET="--enable-static \
-                           --disable-shared \
+                           --enable-shared \
                            --with-ogg=${SYSROOT_PREFIX}/usr \
                            --disable-docs \
                            --disable-examples \
diff --git a/packages/multimedia/ffmpeg/package.mk b/packages/multimedia/ffmpeg/package.mk
index cb0d4ae22a..c457ab97e9 100644
--- a/packages/multimedia/ffmpeg/package.mk
+++ b/packages/multimedia/ffmpeg/package.mk
@@ -8,7 +8,7 @@ PKG_SHA256="57be87c22d9b49c112b6d24bc67d42508660e6b718b3db89c44e47e289137082"
 PKG_LICENSE="GPL-3.0-only"
 PKG_SITE="https://ffmpeg.org"
 PKG_URL="http://ffmpeg.org/releases/ffmpeg-${PKG_VERSION}.tar.xz"
-PKG_DEPENDS_TARGET="toolchain zlib bzip2 openssl speex"
+PKG_DEPENDS_TARGET="toolchain zlib bzip2 openssl speex libvpx libvorbis opus"
 PKG_LONGDESC="FFmpeg is a complete, cross-platform solution to record, convert and stream audio and video."
 PKG_PATCH_DIRS="libreelec"
 
@@ -125,7 +126,7 @@ if [ "${FFMPEG_TESTING}" = "yes" ]; then
     PKG_FFMPEG_TESTING+=" --enable-vout-drm --enable-outdev=vout_drm"
   fi
 else
-  PKG_FFMPEG_TESTING="--disable-programs"
+  PKG_FFMPEG_TESTING=""
 fi
 
 configure_target() {
@@ -188,6 +189,13 @@ configure_target() {
               --enable-encoder=aac \
               --enable-encoder=wmav2 \
               --enable-encoder=mjpeg \
+              --enable-libvpx \
+              --enable-encoder=libvpx_vp8 \
+              --enable-encoder=libvpx_vp9 \
+              --enable-libvorbis \
+              --enable-libopus \
+              --enable-encoder=libvorbis \
+              --enable-encoder=libopus \
               --enable-encoder=png \
               ${PKG_FFMPEG_HWACCEL} \
               --disable-muxers \
@@ -196,6 +205,11 @@ configure_target() {
               --enable-muxer=asf \
               --enable-muxer=ipod \
               --enable-muxer=mpegts \
+              --enable-muxer=matroska \
+              --enable-muxer=mp4 \
+              --enable-muxer=webm \
+              --enable-muxer=webm_chunk \
+              --enable-muxer=webm_dash_manifest \
               --enable-demuxers \
               --enable-parsers \
               --enable-bsfs \
@@ -221,8 +235,6 @@ configure_target() {
               --enable-libspeex \
               --disable-libtheora \
               --disable-libvo-amrwbenc \
-              --disable-libvorbis \
-              --disable-libvpx \
               --disable-libx264 \
               --disable-libxavs \
               --disable-libxvid \
-- 
2.30.2

