From ef7c9a8abe2df1c52e2eabf6044a9cdf37a40664 Mon Sep 17 00:00:00 2001
From: Andreas Baierl <ichgeh@imkreisrum.de>
Date: Fri, 13 Sep 2024 14:14:33 +0200
Subject: [PATCH] attr: fix copying attr includes

Signed-off-by: Andreas Baierl <ichgeh@imkreisrum.de>
---
 packages/devel/attr/package.mk | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/packages/devel/attr/package.mk b/packages/devel/attr/package.mk
index fadb689a84..7949973441 100644
--- a/packages/devel/attr/package.mk
+++ b/packages/devel/attr/package.mk
@@ -26,8 +26,11 @@ fi
 
 makeinstall_target() {
   mkdir -p ${SYSROOT_PREFIX}/usr/lib/
-    cp .libs/libattr.a ${SYSROOT_PREFIX}/usr/lib/
+  cp .libs/libattr.a ${SYSROOT_PREFIX}/usr/lib/
 
   mkdir -p ${SYSROOT_PREFIX}/usr/include/attr
-    cp include/*.h ${SYSROOT_PREFIX}/usr/include/attr
+  for i in `find include/attr/*.h`; do
+    sed -i "s/\<EXPORT\>/extern/" $i
+  done
+  cp include/attr/*.h ${SYSROOT_PREFIX}/usr/include/attr
 }
-- 
2.39.2

