From 0bc824e1e2d238053acf0c753845bc386496226d Mon Sep 17 00:00:00 2001
From: Andreas Baierl <ichgeh@imkreisrum.de>
Date: Thu, 27 Oct 2022 17:07:58 +0200
Subject: [PATCH 1/2] add show_vdr_config option to show_config

---
 config/show_config | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/config/show_config b/config/show_config
index 82029ff957..fac9dcb55e 100644
--- a/config/show_config
+++ b/config/show_config
@@ -142,6 +142,17 @@ show_config() {
     show_distro_config
   fi
 
+  # Distribution specific VDR configuration
+  # show_vdr_distro_config() should be included in one of:
+  #   $DISTRO/show_config
+
+  if [ "${VDR}" != "no" ] && [ -f distributions/${DISTRO}/show_vdr_config ]; then
+    . distributions/${DISTRO}/show_vdr_config
+  fi
+  if [ "${VDR}" != "no" ] && [ "$(type -t show_vdr_distro_config)" = "function" ]; then
+    show_vdr_distro_config
+  fi
+
   config_message+="\n\n ${dashes}${dashes}${dashes}"
   config_message+="\n End Configuration for ${DISTRONAME}"
   config_message+="\n ${dashes}${dashes}${dashes}"
-- 
2.30.2

