diff --git a/scripts/pkgjson b/scripts/pkgjson
index 91b55efaa1..94b1e9e267 100755
--- a/scripts/pkgjson
+++ b/scripts/pkgjson
@@ -5,6 +5,8 @@
 
 . config/options ""
 
+rm -f dont_delete_packages tmp_dont_delete_packages
+
 # This function is passed a list of package.mk paths to be processed.
 # Each package.mk is sourced with relevant variables output in JSON format.
 json_worker() {
@@ -26,6 +28,17 @@ json_worker() {
 
     [[ ${pkgpath} =~ ^${ROOT}/${PACKAGES}/ ]] && hierarchy="global" || hierarchy="local"
 
+    for i in $(echo ${PKG_NEED_UNPACK}| tr " " "\n")
+    do
+        (
+           if [ -f $i/package.mk ]; then
+              . $i/package.mk
+
+              echo $(get_build_dir ${PKG_NAME}) | tee -a tmp_dont_delete_packages >/dev/null
+           fi
+        )
+    done
+
     cat <<EOF
   {
     "name": "${PKG_NAME}",
@@ -39,6 +52,12 @@ json_worker() {
   },
 EOF
   done
+
+
+  if [ -f tmp_dont_delete_packages ]; then
+    sort -u tmp_dont_delete_packages -o dont_delete_packages
+  fi
+
   unset -f exit
 }
 
