diff -ru8bBwd a/packages/devel/spdlog/package.mk b/packages/devel/spdlog/package.mk
--- a/packages/devel/spdlog/package.mk	2022-11-21 15:16:31.534346535 +0100
+++ b/packages/devel/spdlog/package.mk	2022-11-21 15:54:45.532540772 +0100
@@ -11,8 +11,14 @@
 PKG_LONGDESC="Very fast, header only, C++ logging library."
 
 PKG_CMAKE_OPTS_TARGET="-DCMAKE_CXX_STANDARD=14 \
                        -DCMAKE_CXX_EXTENSIONS:BOOL=OFF \
                        -DSPDLOG_BUILD_SHARED=ON \
                        -DSPDLOG_FMT_EXTERNAL=ON \
                        -DSPDLOG_BUILD_EXAMPLE=OFF \
                        -DSPDLOG_BUILD_TESTS=OFF"
+
+post_install() {
+	cd ${INSTALL}/usr/lib
+	ln -s libspdlog.so.${PKG_VERSION} libspdlog.1
+}
+
