diff --git a/projects/Amlogic-ce/packages/addons/driver/dvb-latest/package.mk b/projects/Amlogic-ce/packages/addons/driver/dvb-latest/package.mk
index 70b6843c31..05a44f2e8e 100644
--- a/projects/Amlogic-ce/packages/addons/driver/dvb-latest/package.mk
+++ b/projects/Amlogic-ce/packages/addons/driver/dvb-latest/package.mk
@@ -52,6 +52,8 @@ make_target() {
     sed -e 's/ && RC_CORE//g' -i ${PKG_BUILD}/linux/drivers/media/usb/dvb-usb/Kconfig
   fi
 
+  sed -e 's/CONFIG_DVB_DEMUX_SECTION_LOSS_LOG=y/# CONFIG_DVB_DEMUX_SECTION_LOSS_LOG is not set/g' -i $PKG_BUILD/v4l/.config
+
   # make config all
   kernel_make VER=$KERNEL_VER SRCDIR=$(kernel_path) allyesconfig
 
