diff --git a/packages/network/wireguard-linux-compat/package.mk b/packages/network/wireguard-linux-compat/package.mk
index 34062eccd2..d4898b2822 100644
--- a/packages/network/wireguard-linux-compat/package.mk
+++ b/packages/network/wireguard-linux-compat/package.mk
@@ -4,10 +4,10 @@
 
 PKG_NAME="wireguard-linux-compat"
 PKG_VERSION="v1.0.20220627"
-PKG_SHA256="894f0e0792aa3cc74e93958c175f16ab7155b0049cec940a9000bf7971380f98"
+PKG_SHA256="10aa3890a06fad9c332df6bbab21e05739f48272ea1fe5ac27523c28132ca7e9"
 PKG_LICENSE="GPLv2"
 PKG_SITE="https://www.wireguard.com"
-PKG_URL="https://git.zx2c4.com/wireguard-linux-compat/snapshot/wireguard-linux-compat-$PKG_VERSION.tar.xz"
+PKG_URL="https://git.zx2c4.com/wireguard-linux-compat/snapshot/wireguard-linux-compat-${PKG_VERSION}.zip"
 PKG_DEPENDS_TARGET="toolchain linux libmnl"
 PKG_NEED_UNPACK="$LINUX_DEPENDS"
 PKG_LONGDESC="WireGuard VPN kernel module"
