diff -ru8bBwd a/radiotextosd.c b/radiotextosd.c
--- a/radiotextosd.c	2018-07-15 16:58:31.000000000 +0200
+++ b/radiotextosd.c	2024-10-12 18:46:02.672304572 +0200
@@ -703,17 +703,21 @@
             ct = 0;
             rtp_print();
         }
     }
 
     // check end @ replay
     if (RT_Replay) {
         int rplayCur, rplayTot;
-        cControl::Control()->GetIndex(rplayCur, rplayTot, false);
+        {
+            cMutexLock ControlMutexLock;
+            cControl::Control(ControlMutexLock)->GetIndex(rplayCur, rplayTot, false);
+        }
+
         if (rplayCur >= rplayTot - 1) {
             Hide();
             return osEnd;
         }
     }
 
     // Timeout or no Info/Rass
     if (RT_OsdTO || (RT_OsdTOTemp > 0) || (RT_Info < 0)) {
