diff --git a/packages/devel/pcre2/package.mk b/packages/devel/pcre2/package.mk
index 1b71c0eb55..e5f04e8d33 100644
--- a/packages/devel/pcre2/package.mk
+++ b/packages/devel/pcre2/package.mk
@@ -20,11 +20,13 @@ PKG_CMAKE_OPTS_HOST="-DBUILD_SHARED_LIBS=OFF \
                      -DPCRE2_SUPPORT_JIT=ON \
                      -DPCRE2_BUILD_TESTS=OFF \
                      -DPCRE2_SUPPORT_LIBEDIT=OFF \
+                     -DPCRE2_STATIC_PIC=ON \
                      -DPCRE2_SUPPORT_LIBREADLINE=OFF"
 
 PKG_CMAKE_OPTS_TARGET="-DBUILD_SHARED_LIBS=OFF \
                        -DPCRE2_BUILD_PCRE2_16=ON \
                        -DPCRE2_SUPPORT_LIBEDIT=OFF \
+                       -DPCRE2_STATIC_PIC=ON \
                        -DPCRE2_SUPPORT_LIBREADLINE=OFF"
 
 post_unpack() {
