diff --git a/packages/audio/alsa-utils/package.mk b/packages/audio/alsa-utils/package.mk
index ae1a0cfece..5f449ec2cd 100644
--- a/packages/audio/alsa-utils/package.mk
+++ b/packages/audio/alsa-utils/package.mk
@@ -36,7 +36,7 @@ post_makeinstall_target() {
   rm -rf ${INSTALL}/usr/lib/udev/rules.d/90-alsa-restore.rules
 
   mkdir -p ${INSTALL}/.noinstall
-  for i in aconnect alsamixer amidi aplaymidi arecord arecordmidi aseqdump aseqnet iecset; do
+  for i in aconnect amidi aplaymidi arecord arecordmidi aseqdump aseqnet iecset; do
     mv ${INSTALL}/usr/bin/${i} ${INSTALL}/.noinstall
   done
 
