diff --git a/Makefile b/Makefile
index 90f3261..7445c32 100644
--- a/Makefile
+++ b/Makefile
@@ -36,12 +36,6 @@ APIVERSION = $(call PKGCFG,apiversion)
 
 -include $(PLGCFG)
 
-### export all vars for sub-makes, using absolute paths
-LIBDIR := $(shell cd $(LIBDIR) >/dev/null 2>&1 && pwd)
-LOCDIR := $(shell cd $(LOCDIR) >/dev/null 2>&1 && pwd)
-export
-unexport PLUGIN
-
 ### The name of the distribution archive:
 
 ARCHIVE = $(PLUGIN)-$(VERSION)
@@ -69,21 +63,21 @@ all: client server
 ### Targets:
 
 client:
-	$(MAKE) -C ./tools
+	$(MAKE) LIBDIR=$(LIBDIR) LOCDIR=$(LOCDIR) -C ./tools
 	$(MAKE) -C ./client
 
 server:
-	$(MAKE) -C ./tools
-	$(MAKE) -C ./libdvbmpeg
-	$(MAKE) -C ./remux
-	$(MAKE) -C ./server
+	$(MAKE) LIBDIR=$(LIBDIR) LOCDIR=$(LOCDIR)  -C ./tools
+	$(MAKE) LIBDIR=$(LIBDIR) LOCDIR=$(LOCDIR)  -C ./libdvbmpeg
+	$(MAKE) LIBDIR=$(LIBDIR) LOCDIR=$(LOCDIR)  -C ./remux
+	$(MAKE) LIBDIR=$(LIBDIR) LOCDIR=$(LOCDIR)  -C ./server
 
 install-client: client
-	$(MAKE) -C ./client install
+	$(MAKE) LIBDIR=$(LIBDIR) LOCDIR=$(LOCDIR) APIVERSION=$(APIVERSION) -C ./client install
 	# installs to $(LIBDIR)/libvdr-streamdev-client.so.$(APIVERSION)
 
 install-server: server
-	$(MAKE) -C ./server install
+	$(MAKE) LIBDIR=$(LIBDIR) LOCDIR=$(LOCDIR) APIVERSION=$(APIVERSION) -C ./server install
 	# installs to $(LIBDIR)/libvdr-streamdev-server.so.$(APIVERSION)
 
 install: install-client install-server
-- 
2.30.2

