diff -ru8bBw a/Makefile b/Makefile
--- a/Makefile	2025-01-02 15:28:35.000000000 +0100
+++ b/Makefile	2025-01-08 08:55:55.765819593 +0100
@@ -30,22 +30,16 @@
 ### The version number of VDR's plugin API:
 
 APIVERSION = $(call PKGCFG,apiversion)
 
 ### Allow user defined options to overwrite defaults:
 
 -include $(PLGCFG)
 
-### export all vars for sub-makes, using absolute paths
-LIBDIR := $(shell cd $(LIBDIR) >/dev/null 2>&1 && pwd)
-LOCDIR := $(shell cd $(LOCDIR) >/dev/null 2>&1 && pwd)
-export
-unexport PLUGIN
-
 ### The name of the distribution archive:
 
 ARCHIVE = $(PLUGIN)-$(VERSION)
 PACKAGE = vdr-$(ARCHIVE)
 
 ### Includes and Defines (add further entries here):
 
 INCLUDES += -I$(VDRDIR)/include -I..
@@ -64,30 +58,30 @@
 
 .PHONY: all client server install install-client install-server dist clean
 all: client server
 
 ### Targets:
 
 client:
 	$(Q)$(MAKE) --no-print-directory -C ./tools
-	$(Q)$(MAKE) --no-print-directory -C ./client
+	$(Q)$(MAKE) --no-print-directory LIBDIR=$(LIBDIR) LOCDIR=$(LOCDIR) -C ./tools
 
 server:
-	$(Q)$(MAKE) --no-print-directory -C ./tools
-	$(Q)$(MAKE) --no-print-directory -C ./libdvbmpeg
-	$(Q)$(MAKE) --no-print-directory -C ./remux
-	$(Q)$(MAKE) --no-print-directory -C ./server
+	$(Q)$(MAKE) --no-print-directory LIBDIR=$(LIBDIR) LOCDIR=$(LOCDIR) -C ./tools
+	$(Q)$(MAKE) --no-print-directory LIBDIR=$(LIBDIR) LOCDIR=$(LOCDIR) -C ./libdvbmpeg
+	$(Q)$(MAKE) --no-print-directory LIBDIR=$(LIBDIR) LOCDIR=$(LOCDIR) -C ./remux
+	$(Q)$(MAKE) --no-print-directory LIBDIR=$(LIBDIR) LOCDIR=$(LOCDIR) -C ./server
 
 install-client: client
-	$(Q)$(MAKE) --no-print-directory -C ./client install
+	$(Q)$(MAKE) --no-print-directory LIBDIR=$(LIBDIR) LOCDIR=$(LOCDIR) APIVERSION=$(APIVERSION) -C ./client install
 	$(Q)# installs to $(LIBDIR)/libvdr-streamdev-client.so.$(APIVERSION)
 
 install-server: server
-	$(Q)$(MAKE) --no-print-directory -C ./server install
+	$(Q)$(MAKE) --no-print-directory LIBDIR=$(LIBDIR) LOCDIR=$(LOCDIR) APIVERSION=$(APIVERSION) -C ./server install
 	$(Q)# installs to $(LIBDIR)/libvdr-streamdev-server.so.$(APIVERSION)
 
 install: install-client install-server
 
 dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
 	@mkdir $(TMPDIR)/$(ARCHIVE)
 	@cp -a * $(TMPDIR)/$(ARCHIVE)
