diff --git a/distributions/LibreELEC/options b/distributions/LibreELEC/options
index e1a4a708ae..5e01c9547d 100644
--- a/distributions/LibreELEC/options
+++ b/distributions/LibreELEC/options
@@ -7,7 +7,7 @@
   DESCRIPTION="LibreELEC is a fast and user-friendly Kodi Entertainment Center distribution."
 
 # Distribution Source location
-  DISTRO_MIRROR="https://sources.libreelec.tv/mirror"
+  DISTRO_MIRROR="https://sources.libreelec.tv/mirror https://vdrsternelec.serversenke.de/packages"
   DISTRO_SRC="https://sources.libreelec.tv/$DISTRO_VERSION"
 
 # Distribution Home URL
