diff --git a/packages/devel/cmake/package.mk b/packages/devel/cmake/package.mk
index e0dce3b044..79c98dca87 100644
--- a/packages/devel/cmake/package.mk
+++ b/packages/devel/cmake/package.mk
@@ -3,8 +3,8 @@
 # Copyright (C) 2016-present Team LibreELEC (https://libreelec.tv)
 
 PKG_NAME="cmake"
-PKG_VERSION="4.1.0"
-PKG_SHA256="81ee8170028865581a8e10eaf055afb620fa4baa0beb6387241241a975033508"
+PKG_VERSION="4.0.3"
+PKG_SHA256="8d3537b7b7732660ea247398f166be892fe6131d63cc291944b45b91279f3ffb"
 PKG_LICENSE="BSD"
 PKG_SITE="https://cmake.org/"
 PKG_URL="https://cmake.org/files/v$(get_pkg_version_maj_min)/cmake-${PKG_VERSION}.tar.gz"
