diff --git a/packages/linux/package.mk b/packages/linux/package.mk
index 6a1ed7c2f4..82276ba3d7 100644
--- a/packages/linux/package.mk
+++ b/packages/linux/package.mk
@@ -20,7 +20,7 @@ case "${LINUX}" in
     PKG_SHA256="740368c264d071200577ef745c1e06362564daefd941c7c562905853c6f16598"
     PKG_URL="https://github.com/torvalds/linux/archive/${PKG_VERSION}.tar.gz"
     PKG_SOURCE_NAME="linux-${LINUX}-${PKG_VERSION}.tar.gz"
-    PKG_PATCH_DIRS="default rtlwifi/6.10 rtlwifi/6.11"
+    PKG_PATCH_DIRS="amlogic rtlwifi/6.10 rtlwifi/6.11"
     ;;
   raspberrypi)
     PKG_VERSION="d2813c02131b9ddf938277f4123da7ccbd113ea7" # 6.6.35
