diff -ru8bBwd a/projects/Amlogic-ce/packages/linux/package.mk b/projects/Amlogic-ce/packages/linux/package.mk
--- a/projects/Amlogic-ce/packages/linux/package.mk	2022-12-01 10:39:21.530126893 +0100
+++ b/projects/Amlogic-ce/packages/linux/package.mk	2022-12-01 11:05:51.473185268 +0100
@@ -105,16 +105,19 @@
       sed -e "s|^CONFIG_DRM_LIMA=.*$|# CONFIG_DRM_LIMA is not set|" -i $PKG_BUILD/.config
       sed -e "s|^CONFIG_DRM_PANFROST=.*$|# CONFIG_DRM_PANFROST is not set|" -i $PKG_BUILD/.config
     fi
 
     # disable wireguard support if not enabled
     if [ ! "$WIREGUARD_SUPPORT" = yes ]; then
       sed -e "s|^CONFIG_WIREGUARD=.*$|# CONFIG_WIREGUARD is not set|" -i $PKG_BUILD/.config
     fi
+
+    # enable meson vrtc driver
+    sed -e "s|^# CONFIG_RTC_DRV_MESON_VRTC is not set$|CONFIG_RTC_DRV_MESON_VRTC=m|" -i $PKG_BUILD/.config
   fi
 }
 
 make_host() {
   make \
     ARCH=${HEADERS_ARCH:-$TARGET_KERNEL_ARCH} \
     HOSTCC="$TOOLCHAIN/bin/host-gcc" \
     HOSTCXX="$TOOLCHAIN/bin/host-g++" \
