diff -ru8bBw a/scripts/pkgjson b/scripts/pkgjson
--- a/scripts/pkgjson	2023-01-14 12:31:46.489991288 +0100
+++ b/scripts/pkgjson	2023-01-14 12:32:17.004853396 +0100
@@ -25,16 +25,18 @@
     [ ${exited} -eq 1 ] && continue
 
     if [[ "${pkgpath}" =~ ^${ROOT}/${PACKAGES}/* ]]; then
       hierarchy="global"
     elif [[ "${pkgpath}" =~ ^${ROOT}/projects/${PROJECT}/${PACKAGES}/* ]]; then
       hierarchy="project"
     elif [[ "${pkgpath}" =~ ^${ROOT}/projects/${PROJECT}/devices/${DEVICE}/* ]]; then
       hierarchy="device"
+    else
+      hierarchy="global"
     fi
 
     cat <<EOF
   {
     "name": "${PKG_NAME}",
     "hierarchy": "${hierarchy}",
     "section": "${PKG_SECTION}",
     "bootstrap": "${PKG_DEPENDS_BOOTSTRAP}",
