diff --git a/packages/addons/addon-depends/tntnet/package.mk b/packages/addons/addon-depends/tntnet/package.mk
index 5b94fceb0a..da7b1d59bc 100644
--- a/packages/addons/addon-depends/tntnet/package.mk
+++ b/packages/addons/addon-depends/tntnet/package.mk
@@ -7,7 +7,7 @@ PKG_VERSION="3.0"
 PKG_SHA256="718e5519b0a403f7f766358bf66a85c008119c48189d1c2b7651fd0af9018e27"
 PKG_LICENSE="GPL-2"
 PKG_SITE="http://www.tntnet.org/"
-PKG_URL="http://www.tntnet.org/download/${PKG_NAME}-${PKG_VERSION}.tar.gz"
+PKG_URL="http://slackware.uk/sbosrcarch/by-name/network/tntnet/${PKG_NAME}-${PKG_VERSION}.tar.gz"
 PKG_DEPENDS_HOST="cxxtools:host zlib:host"
 PKG_DEPENDS_TARGET="toolchain tntnet:host libtool cxxtools zlib"
 PKG_LONGDESC="A web application server for C++."
