diff -ru8bBwd a/drivers/amlogic/media/vout/vout_serve/vout_serve.c b/drivers/amlogic/media/vout/vout_serve/vout_serve.c
--- a/drivers/amlogic/media/vout/vout_serve/vout_serve.c	2022-10-05 07:37:37.000000000 +0200
+++ b/drivers/amlogic/media/vout/vout_serve/vout_serve.c	2022-11-09 15:09:35.753155979 +0100
@@ -642,22 +642,21 @@
 
 	return ret;
 }
 
 static ssize_t vout_vinfo_name_show(struct class *class,
 		struct class_attribute *attr, char *buf)
 {
 	const struct vinfo_s *info = NULL;
-	int ret = 0;
 
 	info = get_current_vinfo();
 	if (info == NULL)
 		return sprintf(buf, "null");
-	ret = sprintf(buf, "%s\n", info->name);
+	return sprintf(buf, "%s\n", info->name);
 }
 
 static struct class_attribute vout_class_attrs[] = {
 	__ATTR(mode,       0644, vout_mode_show, vout_mode_store),
 	__ATTR(axis,       0644, vout_axis_show, vout_axis_store),
 	__ATTR(fr_policy,  0644,
 		vout_fr_policy_show, vout_fr_policy_store),
 	__ATTR(fr_hint,    0644, vout_fr_hint_show, vout_fr_hint_store),
@@ -882,17 +881,17 @@
 #ifdef CONFIG_SCREEN_ON_EARLY
 
 	if (early_resume_flag) {
 		early_resume_flag = 0;
 		return 0;
 	}
 
 #endif
-#ifdef CONFIG_AMLOGIC_LEGACY_EARLY_SUSPEND
+#if defined(CONFIG_AMLOGIC_LEGACY_EARLY_SUSPEND) && !defined(CONFIG_AMLOGIC_DRM)
 	if (early_suspend_flag)
 		return 0;
 
 #endif
 	vout_resume();
 	return 0;
 }
 #endif
@@ -929,26 +928,26 @@
 
 	return aml_vout_resume(pdev);
 }
 #endif
 
 #ifdef CONFIG_SCREEN_ON_EARLY
 void resume_vout_early(void)
 {
-#ifdef CONFIG_AMLOGIC_LEGACY_EARLY_SUSPEND
+#if defined(CONFIG_AMLOGIC_LEGACY_EARLY_SUSPEND) && !defined(CONFIG_AMLOGIC_DRM)
 	early_suspend_flag = 0;
 	early_resume_flag = 1;
 	vout_resume();
 #endif
 }
 EXPORT_SYMBOL(resume_vout_early);
 #endif
 
-#ifdef CONFIG_AMLOGIC_LEGACY_EARLY_SUSPEND
+#if defined(CONFIG_AMLOGIC_LEGACY_EARLY_SUSPEND) && !defined(CONFIG_AMLOGIC_DRM)
 static void aml_vout_early_suspend(struct early_suspend *h)
 {
 	if (early_suspend_flag)
 		return;
 
 	vout_suspend();
 	early_suspend_flag = 1;
 }
@@ -1022,17 +1021,17 @@
 static int aml_vout_probe(struct platform_device *pdev)
 {
 	int ret = -1;
 
 	vout_dev = &pdev->dev;
 
 	aml_vout_get_dt_info(pdev);
 
-#ifdef CONFIG_AMLOGIC_LEGACY_EARLY_SUSPEND
+#if defined(CONFIG_AMLOGIC_LEGACY_EARLY_SUSPEND) && !defined(CONFIG_AMLOGIC_DRM)
 	early_suspend.level = EARLY_SUSPEND_LEVEL_BLANK_SCREEN;
 	early_suspend.suspend = aml_vout_early_suspend;
 	early_suspend.resume = aml_vout_late_resume;
 	register_early_suspend(&early_suspend);
 #endif
 
 	vout_class = NULL;
 	ret = vout_attr_create();
@@ -1043,17 +1042,17 @@
 	set_vout_init_mode();
 
 	VOUTPR("%s OK\n", __func__);
 	return ret;
 }
 
 static int aml_vout_remove(struct platform_device *pdev)
 {
-#ifdef CONFIG_AMLOGIC_LEGACY_EARLY_SUSPEND
+#if defined(CONFIG_AMLOGIC_LEGACY_EARLY_SUSPEND) && !defined(CONFIG_AMLOGIC_DRM)
 	unregister_early_suspend(&early_suspend);
 #endif
 
 	aml_vout_extcon_free();
 	vout_attr_remove();
 	vout_fops_remove();
 	vout_unregister_server(&nulldisp_vout_server);
 
