diff -ru8bBwd a/Makefile.since-1.7.35 b/Makefile.since-1.7.35
--- a/Makefile.since-1.7.35	2020-12-09 16:38:01.000000000 +0100
+++ b/Makefile.since-1.7.35	2022-10-30 09:46:34.590680522 +0100
@@ -15,17 +15,17 @@
 BRANCH      = $(shell git rev-parse --abbrev-ref HEAD)
 GIT_REV     = $(shell git describe --always 2>/dev/null)
 
 ### The directory environment:
 
 # Use package data if installed...otherwise assume we're under the VDR source directory:
 PKGCFG = $(if $(VDRDIR),$(shell pkg-config --variable=$(1) $(VDRDIR)/vdr.pc),$(shell pkg-config --variable=$(1) vdr || pkg-config --variable=$(1) ../../../vdr.pc))
 
-LIBDIR = $(call PKGCFG,libdir)
+LIBDIR ?= $(call PKGCFG,libdir)
 LOCDIR = $(call PKGCFG,locdir)
 PLGCFG = $(call PKGCFG,plgcfg)
 
 #
 
 TMPDIR ?= /tmp
 
 ### The compiler options:
@@ -108,17 +108,17 @@
 install-i18n: $(I18Nmsgs)
 
 ### Targets:
 
 $(SOFILE): $(OBJS)
 	$(CXX) $(CXXFLAGS) $(LDFLAGS) -shared $(OBJS) -o $@
 
 install-lib: $(SOFILE)
-	install -D $^ $(DESTDIR)$(LIBDIR)/$^.$(APIVERSION)
+	install -D $^ $(LIBDIR)/$^.$(APIVERSION)
 
 install: install-lib install-i18n
 
 dist: $(I18Npo) clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
 	@mkdir $(TMPDIR)/$(ARCHIVE)
 	@cp -a * $(TMPDIR)/$(ARCHIVE)
 	@tar czf $(PACKAGE).tgz -C $(TMPDIR) --exclude debian --exclude CVS --exclude .svn --exclude .hg --exclude .git $(ARCHIVE)
